<!-- Creator     : groff version 1.19.2 -->
<!-- CreationDate: Thu Jul  4 16:09:43 2019 -->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
"http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<meta name="generator" content="groff -Thtml, see www.gnu.org">
<meta http-equiv="Content-Type" content="text/html; charset=US-ASCII">
<meta name="Content-Style" content="text/css">
<style type="text/css">
       p     { margin-top: 0; margin-bottom: 0; }
       pre   { margin-top: 0; margin-bottom: 0; }
       table { margin-top: 0; margin-bottom: 0; }
</style>
<title>vsearch</title>

</head>
<body>

<h1 align=center>vsearch</h1>

<a href="#NAME">NAME</a><br>
<a href="#SYNOPSIS">SYNOPSIS</a><br>
<a href="#DESCRIPTION">DESCRIPTION</a><br>
<a href="#DELIBERATE CHANGES">DELIBERATE CHANGES</a><br>
<a href="#NOVELTIES">NOVELTIES</a><br>
<a href="#EXAMPLES">EXAMPLES</a><br>
<a href="#AUTHORS">AUTHORS</a><br>
<a href="#CITATION">CITATION</a><br>
<a href="#REPORTING BUGS">REPORTING BUGS</a><br>
<a href="#AVAILABILITY">AVAILABILITY</a><br>
<a href="#COPYRIGHT">COPYRIGHT</a><br>
<a href="#SEE ALSO">SEE ALSO</a><br>
<a href="#VERSION HISTORY">VERSION HISTORY</a><br>

<hr>


<a name="NAME"></a>
<h2>NAME</h2>


<p style="margin-left:11%; margin-top: 1em">vsearch &mdash;
chimera detection, clustering, dereplication and
rereplication, FASTA/FASTQ file processing, masking,
pairwise alignment, searching, shuffling, sorting,
subsampling, and taxonomic classification of amplicons for
metagenomics, genomics, and population genetics.</p>

<a name="SYNOPSIS"></a>
<h2>SYNOPSIS</h2>


<p style="margin-left:11%; margin-top: 1em">Chimera
detection:</p>

<p style="margin-left:22%;"><b>vsearch</b> (--uchime_denovo
| --uchime2_denovo | --uchime3_denovo) <i>fastafile</i>
(--chimeras | --nonchimeras | --uchimealns | --uchimeout)
<i>outputfile</i> [<i>options</i>]</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--uchime_ref <i>fastafile</i> (--chimeras | --nonchimeras |
--uchimealns | --uchimeout) <i>outputfile</i> --db
<i>fastafile</i> [<i>options</i>]</p>


<p style="margin-left:11%; margin-top: 1em">Clustering:</p>

<p style="margin-left:22%;"><b>vsearch</b> (--cluster_fast
| --cluster_size | --cluster_smallmem | --cluster_unoise)
<i>fastafile</i> (--alnout | --biomout | --blast6out |
--centroids | --clusters | --mothur_shared_out | --msaout |
--otutabout | --profile | --samout | --uc | --userout)
<i>outputfile</i> --id <i>real</i> [<i>options</i>]</p>

<p style="margin-left:11%; margin-top: 1em">Dereplication
and rereplication:</p>

<p style="margin-left:22%;"><b>vsearch</b>
(--derep_fulllength | --derep_prefix) <i>fastafile</i>
(--output | --uc) <i>outputfile</i> [<i>options</i>]</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--rereplicate <i>fastafile</i> --output <i>outputfile</i>
[<i>options</i>]</p>

<p style="margin-left:11%; margin-top: 1em">Extraction of
sequences:</p>

<p style="margin-left:22%;"><b>vsearch</b> --fastx_getseq
<i>fastafile</i> (--fastaout | --fastqout | --notmatched |
--notmatchedfq) <i>outputfile</i> --label <i>label</i>
[<i>options</i>]</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--fastx_getseqs <i>fastafile</i> (--fastaout | --fastqout |
--notmatched | --notmatchedfq) <i>outputfile</i> (--label
<i>label &nbsp;</i>--labels <i>labelfile</i> | --label_word
<i>label</i> | --label_words <i>labelfile</i>)
[<i>options</i>]</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--fastx_getsubseq <i>fastafile</i> (--fastaout | --fastqout
| --notmatched | --notmatchedfq) <i>outputfile</i> --label
<i>label</i> [--subseq_start <i>position</i>] [--subseq_end
<i>position</i>] [<i>options</i>]</p>

<p style="margin-left:11%; margin-top: 1em">FASTA/FASTQ
file processing:</p>

<p style="margin-left:22%;"><b>vsearch</b> --fastq_chars
<i>fastqfile</i> [<i>options</i>]</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--fastq_convert <i>fastqfile</i> --fastqout
<i>outputfile</i> [<i>options</i>]</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
(--fastq_eestats | --fastq_eestats2) <i>fastqfile</i>
--output <i>outputfile</i> [<i>options</i>]</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--fastq_filter <i>fastqfile</i> [--reverse <i>fastqfile</i>]
(--fastaout | --fastaout_discarded | --fastqout |
--fastqout_discarded --fastaout_rev |
--fastaout_discarded_rev | --fastqout_rev |
--fastqout_discarded_rev) <i>outputfile</i>
[<i>options</i>]</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--fastq_join <i>fastqfile</i> --reverse <i>fastqfile</i>
(--fastaout | --fastqout) <i>outputfile</i>
[<i>options</i>]</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--fastq_mergepairs <i>fastqfile</i> --reverse
<i>fastqfile</i> (--fastaout | --fastqout |
--fastaout_notmerged_fwd | --fastaout_notmerged_rev |
--fastqout_notmerged_fwd | --fastqout_notmerged_rev |
--eetabbedout) <i>outputfile</i> [<i>options</i>]</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--fastq_stats <i>fastqfile</i> [--log <i>logfile</i>]
[<i>options</i>]</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--fastx_filter <i>inputfile</i> [--reverse <i>inputfile</i>]
(--fastaout | --fastaout_discarded | --fastqout |
--fastqout_discarded --fastaout_rev |
--fastaout_discarded_rev | --fastqout_rev |
--fastqout_discarded_rev) <i>outputfile</i>
[<i>options</i>]</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--fastx_revcomp <i>inputfile</i> (--fastaout | --fastqout)
<i>outputfile</i> [<i>options</i>]</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--sff_convert <i>sff-file</i> --fastqout <i>outputfile</i>
[<i>options</i>]</p>

<p style="margin-left:11%; margin-top: 1em">Masking:</p>

<p style="margin-left:22%;"><b>vsearch</b> --fastx_mask
<i>fastxfile</i> (--fastaout | --fastqout) <i>outputfile</i>
[<i>options</i>]</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--maskfasta <i>fastafile</i> --output <i>outputfile</i>
[<i>options</i>]</p>

<p style="margin-left:11%; margin-top: 1em">Pairwise
alignment:</p>

<p style="margin-left:22%;"><b>vsearch</b>
--allpairs_global <i>fastafile</i> (--alnout | --blast6out |
--matched | --notmatched | --samout | --uc | --userout)
<i>outputfile</i> (--acceptall | --id <i>real</i>)
[<i>options</i>]</p>

<p style="margin-left:11%; margin-top: 1em">Restriction
site cutting:</p>

<p style="margin-left:22%;"><b>vsearch</b> --cut
<i>fastafile</i> --cut_pattern <i>pattern</i> (--fastaout |
--fastaout_rev | --fastaout_discarded |
--fastaout_discarded_rev) <i>outputfile</i>
[<i>options</i>]</p>

<p style="margin-left:11%; margin-top: 1em">Searching:</p>

<p style="margin-left:22%;"><b>vsearch</b> --search_exact
<i>fastafile</i> --db <i>fastafile</i> (--alnout | --biomout
| --blast6out | --mothur_shared_out | --otutabout | --samout
| --uc | --userout) <i>outputfile</i> [<i>options</i>]</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--usearch_global <i>fastafile</i> --db <i>fastafile</i>
(--alnout | --biomout | --blast6out | --mothur_shared_out |
--otutabout | --samout | --uc | --userout) <i>outputfile</i>
--id <i>real</i> [<i>options</i>]</p>

<p style="margin-left:11%; margin-top: 1em">Shuffling and
sorting:</p>

<p style="margin-left:22%;"><b>vsearch</b> (--shuffle |
--sortbylength | --sortbysize) <i>fastafile</i> --output
<i>outputfile</i> [<i>options</i>]</p>


<p style="margin-left:11%; margin-top: 1em">Subsampling:</p>

<p style="margin-left:22%;"><b>vsearch</b>
--fastx_subsample <i>fastafile</i> (--fastaout | --fastqout)
<i>outputfile</i> (--sample_pct <i>real</i> | --sample_size
<i>positive integer</i>) [<i>options</i>]</p>

<p style="margin-left:11%; margin-top: 1em">Taxonomic
classification:</p>

<p style="margin-left:22%;"><b>vsearch</b> --sintax
<i>fastafile</i> --db <i>fastafile</i> --tabbedout
<i>outputfile</i> [--sintax_cutoff <i>real</i>]
[<i>options</i>]</p>

<p style="margin-left:11%; margin-top: 1em">UDB database
handling:</p>

<p style="margin-left:22%;"><b>vsearch</b>
--makeudb_usearch <i>fastafile</i> --output
<i>outputfile</i> [<i>options</i>]</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--udb2fasta <i>udbfile</i> --output <i>outputfile</i>
[<i>options</i>]</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
(--udbinfo | --udbstats) <i>udbfile</i> [<i>options</i>]</p>

<a name="DESCRIPTION"></a>
<h2>DESCRIPTION</h2>


<p style="margin-left:11%; margin-top: 1em">Environmental
or clinical molecular diversity studies generate large
volumes of amplicons (e.g.; SSU-rRNA sequences) that need to
be checked for chimeras, dereplicated, masked, sorted,
searched, clustered or compared to reference sequences. The
aim of <b>vsearch</b> is to offer a all-in-one open source
tool to perform these tasks, using optimized algorithm
implementations and harvesting the full potential of modern
computers, thus providing fast and accurate data
processing.</p>

<p style="margin-left:11%; margin-top: 1em">Comparing
nucleotide sequences is at the core of <b>vsearch</b>. To
speed up comparisons, <b>vsearch</b> implements an extremely
fast Needleman-Wunsch algorithm, making use of the Streaming
SIMD Extensions (SSE2) of post-2003 x86-64 CPUs. If SSE2
instructions are not available, <b>vsearch</b> exits with an
error message. On Power8 CPUs it will use AltiVec/VSX/VMX
instructions. Memory usage increases rapidly with sequence
length: for example comparing two sequences of length 1 kb
requires 8 MB of memory per thread, and comparing two 10 kb
sequences requires 800 MB of memory per thread. For
comparisons involving sequences with a length product
greater than 25 million (for example two sequences of length
5 kb), <b>vsearch</b> uses a slower alignment method
described by Hirschberg (1975) and Myers and Miller (1988),
with much smaller memory requirements.</p>

<p style="margin-left:11%; margin-top: 1em"><b>Input <br>
vsearch</b> accept as input fasta or fastq files containing
one or several nucleotidic entries. In fasta files, each
nucleotidic entry is made of a header and a sequence. The
header is defined as the string comprised between the
&rsquo;&gt;&rsquo; symbol and the first space, tab or the
end of the line, whichever comes first. Additionally, if the
header matches <i>integer</i> as the number of occurrences
(or abundance) of the sequence in the study. That abundance
information is used or created during chimera detection,
clustering, dereplication, sorting and searching.</p>

<p style="margin-left:11%; margin-top: 1em">The sequence is
defined as a string of IUPAC symbols (ACGTURYSWKMDBHVN),
starting after the end of the identifier line and ending
before the next identifier line, or the file end.
<b>vsearch</b> silently ignores ascii characters 9 to 13,
and exits with an error message if ascii characters 0 to 8,
14 to 31, &rsquo;.&rsquo; or &rsquo;-&rsquo; are present.
All other ascii or non-ascii characters are stripped and
complained about in a warning message.</p>

<p style="margin-left:11%; margin-top: 1em">In fastq files,
each entry is made of sequence header starting with a symbol
&rsquo;@&rsquo;, a nucleotidic sequence (same rules as for
fasta sequences), a quality header starting with a symbol
&rsquo;+&rsquo; and a string of ASCII characters (offset 33
or 64), each one encoding the quality value of the
corresponding position in the nucleotidic sequence.</p>

<p style="margin-left:11%; margin-top: 1em"><b>vsearch</b>
operations are case insensitive, except when soft masking is
activated. Masking is automatically applied during chimera
detection, clustering, masking, pairwise alignment and
searching. Soft masking is specified with the options
&rsquo;--dbmask soft&rsquo; (for searching and chimera
detection with a reference) or &rsquo;--qmask soft&rsquo;
(for searching, <i>de novo</i> chimera detection, clustering
and masking). When using soft masking, lower case letters
indicate masked symbols, while upper case letters indicate
regular symbols. Masked symbols are never included in the
unique index words used for sequence comparisons, otherwise
they are treated as normal symbols.</p>

<p style="margin-left:11%; margin-top: 1em">When comparing
sequences during chimera detection, dereplication, searching
and clustering, T and U are considered identical, regardless
of their case. When aligning sequences, identical symbols
will receive a positive match score (default +2). If two
symbols are not identical, their alignment result in a
negative mismatch score (default -4). Aligning a pair of
symbols where at least one of them is an ambiguous symbol
(BDHKMNRSVWY) will always result in a score of zero.
Alignment of two identical ambiguous symbols (for example, R
vs R) also receives a score of zero. When computing the
amount of similarity by counting matches and mismatches
after alignment, ambiguous nucleotide symbols will count as
matching to other symbols if they have at least one of the
nucleotides (ACGTU) they may represent in common. For
example: W will match A and T, but also any of MRVHDN. When
showing alignments (for example with the --alnout option)
matches involving ambiguous symbols will be shown with a
plus character (+) between them while exact matches between
non-ambiguous symbols will be shown with a vertical bar
character (|).</p>

<p style="margin-left:11%; margin-top: 1em"><b>vsearch</b>
can read data from standard files and write to standard
files, but it can also read from pipes and write to pipes!
For example, multiple fasta files can be piped into
<b>vsearch</b> for dereplication. To do so, file names can
be replaced with:</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top">the symbol
&rsquo;-&rsquo;, representing &rsquo;/dev/stdin&rsquo; for
input files or &rsquo;/dev/stdout&rsquo; for output
files,</p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top">a named pipe
created with the command mkfifo,</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top">a process
substitution &rsquo;&lt;(command)&rsquo; as input or
&rsquo;&gt;(command)&rsquo; as output.</p></td>
</table>

<p style="margin-left:11%; margin-top: 1em"><b>vsearch</b>
can automatically read compressed gzip or bzip2 files if the
appropriate libraries are present during the compilation.
<b>vsearch</b> can also read pipes streaming compressed gzip
or bzip2 data if the options --gzip_decompress or
--bzip2_decompress are selected. When reading from a pipe,
the progress indicator is not updated.</p>

<p style="margin-left:11%; margin-top: 1em"><b>Options <br>
vsearch</b> recognizes a large number of command-line
commands and options. For easier navigation, options are
grouped below by theme (chimera detection, clustering,
dereplication and rereplication, FASTA/FASTQ file
processing, masking, pairwise alignment, searching,
shuffling, sorting, and subsampling). We start with the
general options that apply to all themes. Options start with
a double dash (--). A single dash (-) may also be used,
except on NetBSD systems. Option names may be shortened as
long as they are not ambiguous (e.g. --derep_f).</p>

<p style="margin-left:11%; margin-top: 1em">Help and
version commands:</p>


<p style="margin-left:22%; margin-top: 1em"><a name="help"></a>
<a name="h"></a> <b><br>
--help --h</b></p>

<p style="margin-left:35%;">Display help text with brief
information about all commands and options.
<a name="version"></a> <a name="v"></a></p>

<p style="margin-left:22%;"><b>--version --v</b></p>

<p style="margin-left:35%;">Output version information and
a citation for the VSEARCH publication. Show the status of
the support for gzip- and bzip2-compressed input files.</p>

<p style="margin-left:11%; margin-top: 1em">General
options:</p>


<p style="margin-left:22%;"><a name="bzip2_decompress"></a>
<b><br>
--bzip2_decompress</b></p>

<p style="margin-left:35%;">When reading from a pipe
streaming bzip2-compressed data, decompress the data. That
option is not needed when reading from a standard
bzip2-compressed file. <a name="fasta_width"></a></p>


<p style="margin-left:22%;"><b>--fasta_width&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Fasta files produced by
<b>vsearch</b> are wrapped (sequences are written on lines
of <i>integer</i> nucleotides, 80 by default). Set that
value to zero to eliminate the wrapping.
<a name="gzip_decompress"></a></p>

<p style="margin-left:22%;"><b>--gzip_decompress</b></p>

<p style="margin-left:35%;">When reading from a pipe
streaming gzip-compressed data, decompress the data. That
option is not needed when reading from a standard
gzip-compressed file. <a name="log"></a></p>

<p style="margin-left:22%;"><b>--log</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write messages to the specified
log file. Information written includes program version,
amount of memory available, number of cores and command line
options, and if need be, informational messages, warnings
and fatal errors. The start and finish times are also
recorded as well as the elapsed time and the maximum amount
of memory consumed. The different <b>vsearch</b> commands
can also write additional informations to the log file.
<a name="maxseqlength"></a></p>


<p style="margin-left:22%;"><b>--maxseqlength&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">All <b>vsearch</b> operations
discard sequences of length equal or greater than
<i>integer</i> (50,000 nucleotides by default).
<a name="minseqlength"></a></p>


<p style="margin-left:22%;"><b>--minseqlength&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">All <b>vsearch</b> operations
discard sequences of length smaller than <i>integer</i>: 1
nucleotide by default for sorting or shuffling, 32
nucleotides for clustering, dereplication or searching.
<a name="no_progress"></a></p>

<p style="margin-left:22%;"><b>--no_progress</b></p>

<p style="margin-left:35%;">Do not show the gradually
increasing progress indicator.
<a name="notrucnlabels"></a></p>

<p style="margin-left:22%;"><b>--notrunclabels</b></p>

<p style="margin-left:35%;">Do not truncate sequence labels
at first space or tab, use the full header in output files.
<a name="quiet"></a></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>--quiet</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Suppress all
messages to stdout and stderr except for warnings and fatal
error messages. <a name="threads"></a></p></td>
</table>


<p style="margin-left:22%;"><b>--threads&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Number of computation threads
to use (1 to 1024). The number of threads should be lesser
or equal to the number of available CPU cores. The default
is to use all available resources and to launch one thread
per logical core. The following commands are multi-threaded:
allpairs_global, cluster_fast, cluster_size,
cluster_smallmem, cluster_unoise, fastq_mergepairs,
fastx_mask, maskfasta, search_exact, sintax, uchime_ref, and
usearch_global. Only one thread is used for the other
commands.</p>

<p style="margin-left:11%; margin-top: 1em">Chimera
detection options:</p>

<p style="margin-left:22%; margin-top: 1em">Chimera
detection is based on a scoring function controlled by five
options (--dn, --mindiffs, --mindiv, --minh, --xn).
Sequences are first sorted by decreasing abundance, if
available, and compared on their <i>plus</i> strand only
(case insensitive).</p>

<p style="margin-left:22%; margin-top: 1em">Input sequences
are masked as specified with the --qmask and --hardmask
options. Masking of the database for reference based chimera
detection is specified with the --dbmask option.</p>

<p style="margin-left:22%; margin-top: 1em">In <i>de
novo</i> mode, input fasta file should present abundance
annotations (i.e. a pattern [;]size=<i>integer</i>[;] in the
fasta header). Input order matters for chimera detection, so
we recommend to sort sequences by decreasing abundance
(default of --derep_fulllength command). If your sequence
set needs to be sorted, please see the --sortbysize command
in the sorting section.</p>


<p style="margin-left:22%; margin-top: 1em"><a name="abskew"></a>
<b><br>
--abskew</b> <i>real</i></p>

<p style="margin-left:35%;">When using --uchime_denovo, the
abundance skew is used to distinguish in a three-way
alignment which sequence is the chimera and which are the
parents. The assumption is that chimeras appear later in the
PCR amplification process and are therefore less abundant
than their parents. For --uchime3_denovo the default value
is 16.0. For the other commands, the default value is 2.0,
which means that the parents should be at least 2 times more
abundant than their chimera. Any positive value equal or
greater than 1.0 can be used. <a name="alignwidth"></a></p>


<p style="margin-left:22%;"><b>--alignwidth&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --uchimealns, set
the width of the three-way alignments (80 nucleotides by
default). Set to zero to eliminate wrapping.
<a name="borderline"></a></p>

<p style="margin-left:22%;"><b>--borderline</b>
<i>filename</i></p>

<p style="margin-left:35%;">Output borderline chimeric
sequences to <i>filename</i>, in fasta format. Borderline
chimeric sequences are sequences that have a high enough
score but which are not sufficiently different from their
closest parent. <a name="chimeras"></a></p>

<p style="margin-left:22%;"><b>--chimeras</b>
<i>filename</i></p>

<p style="margin-left:35%;">Output chimeric sequences to
<i>filename</i>, in fasta format. Output order may vary when
using multiple threads. <a name="db"></a></p>

<p style="margin-left:22%;"><b>--db</b> <i>filename</i></p>

<p style="margin-left:35%;">When using --uchime_ref, detect
chimeras using the fasta-formatted reference sequences
contained in <i>filename</i>. Reference sequences are
assumed to be chimera-free. Chimeras cannot be detected if
their parents, or sufficiently close relatives, are not
present in the database. <a name="dn"></a></p>

<p style="margin-left:22%;"><b>--dn</b> <i>real</i></p>

<p style="margin-left:35%;">No vote pseudo-count,
corresponding to the parameter <i>n</i> in the chimera
scoring function (default value is 1.4).
<a name="fasta_score"></a></p>

<p style="margin-left:22%;"><b>--fasta_score</b></p>

<p style="margin-left:35%;">Add the chimera score to the
headers in the fasta output files for chimeras, non-chimeras
and borderline sequences, using the format
<a name="mindiffs"></a></p>


<p style="margin-left:22%;"><b>--mindiffs&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Minimum number of differences
per segment (default value is 3). The parameter is ignored
with --uchime2_denovo and --uchime3_denovo.
<a name="mindiv"></a></p>

<p style="margin-left:22%;"><b>--mindiv</b> <i>real</i></p>

<p style="margin-left:35%;">Minimum divergence from closest
parent (default value is 0.8). The parameter is ignored with
--uchime2_denovo and --uchime3_denovo.
<a name="minh"></a></p>

<p style="margin-left:22%;"><b>--minh</b> <i>real</i></p>

<p style="margin-left:35%;">Minimum score (<i>h</i>).
Increasing this value tends to reduce the number of false
positives and to decrease sensitivity. Default value is
0.28, and values ranging from 0.0 to 1.0 included are
accepted. The parameter is ignored with --uchime2_denovo and
--uchime3_denovo. <a name="nonchimeras"></a></p>

<p style="margin-left:22%;"><b>--nonchimeras</b>
<i>filename</i></p>

<p style="margin-left:35%;">Output non-chimeric sequences
to <i>filename</i>, in fasta format. Output order may vary
when using multiple threads. <a name="relabel"></a></p>

<p style="margin-left:22%;"><b>--relabel</b>
<i>string</i></p>

<p style="margin-left:35%;">Relabel sequences using the
prefix <i>string</i> and a ticker (1, 2, 3, etc.) to
construct the new headers. Use --sizeout to conserve the
abundance annotations. <a name="relabel_keep"></a></p>

<p style="margin-left:22%;"><b>--relabel_keep</b></p>

<p style="margin-left:35%;">When relabelling, keep the old
identifier in the header after a space.
<a name="relabel_md5"></a></p>

<p style="margin-left:22%;"><b>--relabel_md5</b></p>

<p style="margin-left:35%;">Relabel sequences using the MD5
message digest algorithm applied to each sequence. Former
sequence headers are discarded. The sequence is converted to
upper case and each &rsquo;U&rsquo; is replaced by a
&rsquo;T&rsquo; before computation of the digest. The MD5
digest is a cryptographic hash function designed to minimize
the probability that two different inputs give the same
output, even for very similar, but non-identical inputs.
Still, there is a very small, but non-zero, probability that
two different inputs give the same digest (i.e. a
collision). MD5 generates a 128-bit (16-byte) digest that is
represented by 16 hexadecimal numbers (using 32 symbols
among 0123456789abcdef). Use --sizeout to conserve the
abundance annotations. <a name="relabel_sha1"></a></p>

<p style="margin-left:22%;"><b>--relabel_sha1</b></p>

<p style="margin-left:35%;">Relabel sequences using the
SHA1 message digest algorithm applied to each sequence. It
is similar to the --relabel_md5 option but uses the SHA1
algorithm instead of the MD5 algorithm. SHA1 generates a
160-bit (20-byte) digest that is represented by 20
hexadecimal numbers (40 symbols). The probability of a
collision (two non-identical sequences resulting in the same
digest) is smaller for the SHA1 algorithm than it is for the
MD5 algorithm. <a name="self"></a></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="22%"></td>
<td width="12%">


<p style="margin-top: 1em" valign="top"><b>--self</b></p></td>
<td width="1%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">When using
--uchime_ref, ignore a reference sequence when its label
matches the label of the query sequence (useful to estimate
false-positive rate in reference sequences).
<a name="selfid"></a></p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="12%">



<p style="margin-top: 1em" valign="top"><b>--selfid</b></p> </td>
<td width="1%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">When using
--uchime_ref, ignore a reference sequence when its
nucleotide sequence is strictly identical to the nucleotidic
sequence of the query. <a name="sizeout"></a></p></td>
</table>

<p style="margin-left:22%;"><b>--sizeout</b></p>

<p style="margin-left:35%;">When relabelling, add abundance
annotations to fasta headers (using the format
&rsquo;;size=<i>integer</i>;&rsquo;).
<a name="uchime_denovo"></a></p>

<p style="margin-left:22%;"><b>--uchime_denovo</b>
<i>filename</i></p>

<p style="margin-left:35%;">Detect chimeras present in the
fasta-formatted <i>filename</i>, without external references
(i.e. <i>de novo</i>). Automatically sort the sequences in
<i>filename</i> by decreasing abundance beforehand (see the
sorting section for details). Multithreading is not
supported. <a name="uchime2_denovo"></a></p>

<p style="margin-left:22%;"><b>--uchime2_denovo</b>
<i>filename</i></p>

<p style="margin-left:35%;">Detect chimeras present in the
fasta-formatted <i>filename</i>, using the UCHIME2
algorithm. This algorithm is designed for denoised amplicons
(see --cluster_unoise). Automatically sort the sequences in
<i>filename</i> by decreasing abundance beforehand (see the
sorting section for details). Multithreading is not
supported. <a name="uchime3_denovo"></a></p>

<p style="margin-left:22%;"><b>--uchime3_denovo</b>
<i>filename</i></p>

<p style="margin-left:35%;">Detect chimeras present in the
fasta-formatted <i>filename</i>, using the UCHIME2
algorithm. The only difference from --uchime2_denovo is that
the default minimum abundance skew (--abskew) is set to 16.0
rather than 2.0. <a name="uchime_ref"></a></p>

<p style="margin-left:22%;"><b>--uchime_ref</b>
<i>filename</i></p>

<p style="margin-left:35%;">Detect chimeras present in the
fasta-formatted <i>filename</i> by comparing them with
reference sequences (option --db). Multithreading is
supported. <a name="uchimealns"></a></p>

<p style="margin-left:22%;"><b>--uchimealns</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write the three-way global
alignments (parentA, parentB, chimera) to <i>filename</i>
using a human-readable format. Use --alignwidth to modify
alignment length. Output order may vary when using multiple
threads. All sequences are converted to upper case before
alignment. Lower case letters indicate disagreement in the
alignment. <a name="uchimeout"></a></p>

<p style="margin-left:22%;"><b>--uchimeout</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write chimera detection results
to <i>filename</i> using a 18-field, tab-separated
uchime-like format. Use --uchimeout5 to use a format
compatible with usearch v5 and earlier versions. Rows output
order may vary when using multiple threads.</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">1.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">score: higher score
means a more likely chimeric alignment.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">2.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Q: query sequence
label.</p> </td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">3.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">A: parent A
sequence label.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">4.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">B: parent B
sequence label.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">5.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">T: top parent
sequence label (i.e. parent most similar to the query). That
field is removed when using --uchimeout5.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">6.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">idQM: percentage of
similarity of query (Q) and model (M) constructed as a part
of parent A and a part of parent B.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">7.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">idQA: percentage of
similarity of query (Q) and parent A.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">8.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">idQB: percentage of
similarity of query (Q) and parent B.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">9.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">idAB: percentage of
similarity of parent A and parent B.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">10.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">idQT: percentage of
similarity of query (Q) and top parent (T).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">11.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">LY: yes votes in
the left part of the model.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">12.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">LN: no votes in the
left part of the model.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">13.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">LA: abstain votes
in the left part of the model.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">14.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">RY: yes votes in
the right part of the model.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">15.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">RN: no votes in the
right part of the model.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">16.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">RA: abstain votes
in the right part of the model.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">17.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">div: divergence,
defined as (idQM - idQT).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">18.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">YN: query is
chimeric (Y), or not (N), or is a borderline case (?).</p></td>
</table>

<p style="margin-left:22%;"><a name="uchimeout5"></a>
<b><br>
--uchimeout5</b></p>

<p style="margin-left:35%;">When using --uchimeout, write
chimera detection results using a 17-field, tab-separated
uchime-like format (drop the 5th field of --uchimeout),
compatible with usearch version 5 and earlier versions.
<a name="xn"></a></p>

<p style="margin-left:22%;"><b>--xn</b> <i>real</i></p>

<p style="margin-left:35%;">No vote weight, corresponding
to the parameter <i>beta</i> in the scoring function
(default value is 8.0). <a name="xsize"></a></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>--xsize</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Strip abundance
information from the headers when writing the output
file.</p> </td>
</table>

<p style="margin-left:11%; margin-top: 1em">Clustering
options:</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
implements a single-pass, greedy centroid-based clustering
algorithm, similar to the algorithms implemented in usearch,
DNAclust and sumaclust for example. Important parameters are
the global clustering threshold (--id) and the pairwise
identity definition (--iddef).</p>

<p style="margin-left:22%; margin-top: 1em">Input sequences
are masked as specified with the --qmask and --hardmask
options. <a name="biomout"></a> <b><br>
--biomout</b> <i>filename</i></p>

<p style="margin-left:35%;">Generate an OTU table in the
biom version 1.0 JSON file format as specified at
<a href="http://biom-format.org/documentation/format_versions/biom-1.0.html">(link)</a>
&lt;http://biom-format.org/documentation/format_versions/biom-1.0.html&gt;.
The format describes how to store a sparse matrix containing
the abundances of the OTUs in the different samples. This
format is much more efficient than the classic and mothur
OTU table formats available with the --otutabout and
--mothur_shared_out options, respectively, and is
recommended at least for large tables. The OTUs are
represented by the cluster centroids. Taxonomy information
will be included for the OTUs if available. Sample
identifiers will be extracted from the headers of all
sequences in the input file. If the header contains
&rsquo;;sample=abc123;&rsquo; or
&rsquo;;barcodelabel=abc123;&rsquo; or a similar string
somewhere, then the given sample identifier (here
&rsquo;abc123&rsquo;) will be used. The semicolon is not
mandatory at the beginning or end of the header. The sample
identifier may contain any printable character except
semicolons. If no such sample label is found, the identifier
in the initial part of the header will be used, but only
letters, digits and underscores are allowed. OTU identifiers
will be extracted from the headers of the cluster centroid
sequences. If the header contains &rsquo;;otu=def789;&rsquo;
or a similar string somewhere, then the given OTU identifier
(here &rsquo;def789&rsquo;) will be used. The semicolon is
not mandatory at the beginning or end of the header. The OTU
identifier may contain any printable character except
semicolons. If no such OTU label is found, the identifier in
the initial part of the header will be used, and all
characters except semicolons are allowed. Alternatively, OTU
identifers can be generated using the relabelling options
(--relabel, --relabel_sha1 or --relabel_md5). Taxonomy
information, if present, will also be extracted from the
headers of the centroid sequences. If the header contains
&rsquo;;tax=Homo_sapiens;&rsquo; or a similar string
somewhere, then the given taxonomy information (here
&rsquo;Homo_sapiens&rsquo;) will be used. The semicolon is
not mandatory at the beginning or end of the header. The
taxonomy information may contain any printable character
except semicolons. If an OTU table in the biom version 2.1
HDF5 file format is required, the biom utility may be used
as described at
<a href="http://biom-format.org/documentation/biom_conversion.html">(link)</a>
&lt;http://biom-format.org/documentation/biom_conversion.html&gt;.
<a name="centroids"></a></p>

<p style="margin-left:22%;"><b>--centroids</b>
<i>filename</i></p>

<p style="margin-left:35%;">Output cluster centroid
sequences to <i>filename</i>, in fasta format. The centroid
is the sequence that seeded the cluster (i.e. the first
sequence of the cluster). <a name="clusterout_id"></a></p>

<p style="margin-left:22%;"><b>--clusterout_id</b></p>

<p style="margin-left:35%;">Add cluster identifier
information to the output files when using the --consout and
--profile options. <a name="clusterout_sort"></a></p>

<p style="margin-left:22%;"><b>--clusterout_sort</b></p>

<p style="margin-left:35%;">Sort output files by decreasing
abundance when using the --consout, --msaout and --profile
options. <a name="cluster_fast"></a></p>

<p style="margin-left:22%;"><b>--cluster_fast</b>
<i>filename</i></p>

<p style="margin-left:35%;">Clusterize the fasta sequences
in <i>filename</i>, automatically sort by decreasing
sequence length beforehand. <a name="cluster_size"></a></p>

<p style="margin-left:22%;"><b>--cluster_size</b>
<i>filename</i></p>

<p style="margin-left:35%;">Clusterize the fasta sequences
in <i>filename</i>, automatically sort by decreasing
sequence abundance beforehand.
<a name="cluster_smallmem"></a></p>

<p style="margin-left:22%;"><b>--cluster_smallmem</b>
<i>filename</i></p>

<p style="margin-left:35%;">Clusterize the fasta sequences
in <i>filename</i> without automatically modifying their
order beforehand. Sequence are expected to be sorted by
decreasing sequence length, unless --usersort is used.
<a name="cluster_unoise"></a></p>

<p style="margin-left:22%;"><b>--cluster_unoise</b>
<i>filename</i></p>

<p style="margin-left:35%;">Perform denoising of the fasta
sequences in <i>filename</i> according to the UNOISE version
3 algorithm by Robert Edgar, but without the chimera removal
step. The options --minsize (default 8) and --unoise_alpha
(default 2.0) may be specified. Chimera removal (<i>de
novo</i>) should be performed afterwards with
--uchime3_denovo. <a name="clusters"></a></p>

<p style="margin-left:22%;"><b>--clusters</b>
<i>string</i></p>

<p style="margin-left:35%;">Output each cluster to a
separate fasta file using the prefix <i>string</i> and a
ticker (0, 1, 2, etc.) to construct the path and filenames.
<a name="consout"></a></p>

<p style="margin-left:22%;"><b>--consout</b>
<i>filename</i></p>

<p style="margin-left:35%;">Output cluster consensus
sequences to <i>filename</i>. For each cluster, a multiple
alignment is computed, and a consensus sequence is
constructed by taking the majority symbol (nucleotide or
gap) from each column of the alignment. Columns containing a
majority of gaps are skipped, except for terminal gaps. If
the --sizein option is specified, sequence abundances will
be taken into account. <a name="cons_truncate"></a></p>

<p style="margin-left:22%;"><b>--cons_truncate</b></p>

<p style="margin-left:35%;">This command is ignored. A
warning is issued. <a name="id"></a></p>

<p style="margin-left:22%;"><b>--id</b> <i>real</i></p>

<p style="margin-left:35%;">Do not add the target to the
cluster if the pairwise identity with the centroid is lower
than <i>real</i> (value ranging from 0.0 to 1.0 included).
The pairwise identity is defined as the number of (matching
columns) / (alignment length - terminal gaps). That
definition can be modified by --iddef.
<a name="iddef"></a></p>


<p style="margin-left:22%;"><b>--iddef&nbsp;</b><i>0|1|2|3|4</i></p>

<p style="margin-left:35%;">Change the pairwise identity
definition used in --id. Values accepted are:</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">0.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">CD-HIT definition:
(matching columns) / (shortest sequence length).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">1.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">edit distance:
(matching columns) / (alignment length).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">2.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">edit distance
excluding terminal gaps (same as --id).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">3.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Marine Biological
Lab definition counting each gap opening (internal or
terminal) as a single mismatch, whether or not the gap was
extended: 1.0 - [(mismatches + gap openings)/(longest
sequence length)]</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">4.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">BLAST definition,
equivalent to --iddef 1 in a context of global pairwise
alignment.</p> </td>
</table>

<p style="margin-left:22%;"><a name="minsize"></a> <b><br>
--minsize&nbsp;</b><i>positive integer</i></p>

<p style="margin-left:35%;">Specify the minimum abundance
of sequences for denoising using --cluster_unoise. The
default is 8. <a name="msaout"></a></p>

<p style="margin-left:22%;"><b>--msaout</b>
<i>filename</i></p>

<p style="margin-left:35%;">Output a multiple sequence
alignment and a consensus sequence for each cluster to
<i>filename</i>, in fasta format. Be warned that vsearch
computes center star multiple sequence alignments using a
fast method whose accuracy can decrease significantly when
using low pairwise identity thresholds. The consensus
sequence is constructed by taking the majority symbol
(nucleotide or gap) from each column of the alignment.
Columns containing a majority of gaps are skipped, except
for terminal gaps. If the --sizein option is specified,
sequence abundances will be taken into account when
computing the consensus.
<a name="mothur_shared_out"></a></p>

<p style="margin-left:22%;"><b>--mothur_shared_out</b>
<i>filename</i></p>

<p style="margin-left:35%;">Output an OTU table in the
mothur &rsquo;shared&rsquo; tab-separated plain text format
as described at
<a href="https://www.mothur.org/wiki/Shared_file">(link)</a>
&lt;https://www.mothur.org/wiki/Shared_file&gt;. The format
describes how a matrix containing the abundances of the OTUs
in the different samples is stored. The first line will
start with the strings &rsquo;label&rsquo;,
&rsquo;group&rsquo; and &rsquo;numOtus&rsquo; and is
followed by a list of all OTU identifiers. The following
lines, one for each sample, starts with the string
&rsquo;vsearch&rsquo; followed by the sample identifier, the
total number of OTUs, and a list of abundances for each OTU
in that sample, in the order given on the first line. The
OTU and sample identifiers are extracted from the FASTA
headers of the sequences. The OTUs are represented by the
cluster centroids. See the --biomout option for further
details. <a name="otutabout"></a></p>

<p style="margin-left:22%;"><b>--otutabout</b>
<i>filename</i></p>

<p style="margin-left:35%;">Output an OTU table in the
classic tab-separated plain text format as a matrix
containing the abundances of the OTUs in the different
samples. The first line will start with the string
&rsquo;#OTU ID&rsquo; and is followed by a tab-separated
list of all sample identifiers. The following lines, one for
each OTU, starts with the OTU identifier and is followed by
a tab-separated list of abundances for that OTU in each
sample, in the order given on the first line. The OTU and
sample identifiers are extracted from the FASTA headers of
the sequences. The OTUs are represented by the cluster
centroids. An extra column is added to the right of the
table if taxonomy information is available for at least one
of the OTUs. This column will be labelled
&rsquo;taxonomy&rsquo; and each row will then contain the
taxonomy information extracted for that OTU. See the
--biomout option for further details.
<a name="profile"></a></p>

<p style="margin-left:22%;"><b>--profile</b>
<i>filename</i></p>

<p style="margin-left:35%;">Output a sequence profile to a
text file with the frequency of each nucleotide in each
position in the multiple alignment for each cluster. There
is a FASTA-like header line for each cluster, followed by
the profile information in a tab-separated format. The eight
columns are: position (0-based), consensus nucleotide,
number of As, number of Cs, number of Gs, number of Ts or
Us, number of gap symbols, and finally the total number of
ambiguous nucleotide symbols (B, D, H, K, M, N, R, S, Y, V
or W). All numbers are integers. If the --sizein option is
specified, sequence abundances will be taken into account.
<a name="qmask"></a></p>


<p style="margin-left:22%;"><b>--qmask&nbsp;</b><i>none|dust|soft</i></p>

<p style="margin-left:35%;">Mask regions in sequences using
the <i>dust</i> or the <i>soft</i> methods, or do not mask
(<i>none</i>). Warning, when using <i>soft</i> masking,
clustering becomes case sensitive. The default is to mask
using <i>dust</i>. <a name="relabel"></a></p>

<p style="margin-left:22%;"><b>--relabel</b>
<i>string</i></p>

<p style="margin-left:35%;">Relabel sequence identifiers in
the output files produced by --consout, --profile and
--centroids options. Please see the description of the same
option under Chimera detection for details.
<a name="relabel_keep"></a></p>

<p style="margin-left:22%;"><b>--relabel_keep</b></p>

<p style="margin-left:35%;">When relabelling, keep the old
identifier in the header after a space.
<a name="relabel_md5"></a></p>

<p style="margin-left:22%;"><b>--relabel_md5</b></p>

<p style="margin-left:35%;">Relabel sequence identifiers in
the output files produced by --consout, --profile and
--centroids options. Please see the description of the same
option under Chimera detection for details.
<a name="relabel_sha1"></a></p>

<p style="margin-left:22%;"><b>--relabel_sha1</b></p>

<p style="margin-left:35%;">Relabel sequence identifiers in
the output files produced by --consout, --profile and
--centroids options. Please see the description of the same
option under Chimera detection for details.
<a name="sizein"></a></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="22%"></td>
<td width="12%">



<p style="margin-top: 1em" valign="top"><b>--sizein</b></p> </td>
<td width="1%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Take into account
the abundance annotations present in the input fasta file
(search for the pattern
&rsquo;[&gt;;]size=<i>integer</i>[;]&rsquo; in sequence
headers). <a name="sizeorder"></a></p></td>
</table>

<p style="margin-left:22%;"><b>--sizeorder</b></p>

<p style="margin-left:35%;">When an amplicon is close to 2
or more centroids, both within the distance specified with
the --id option, resolve the ambiguity by clustering it with
the centroid having the highest abundance, not necessarily
the closest one. The option only has effect when the value
specified with --maxaccepts is higher than one. The
--sizeorder option turns on what is sometimes referred to as
abundance-based greedy clustering (AGC), in contrast to the
default distance-based greedy clustering (DGC).
<a name="sizeout"></a></p>

<p style="margin-left:22%;"><b>--sizeout</b></p>

<p style="margin-left:35%;">Add abundance annotations to
the output fasta files (add the pattern specified, abundance
annotations are reported to output files, and each cluster
centroid receives a new abundance value corresponding to the
total abundance of the amplicons included in the cluster
(--centroids option). If --sizein is not specified, input
abundances are set to 1 for amplicons, and to the number of
amplicons per cluster for centroids.
<a name="strand"></a></p>


<p style="margin-left:22%;"><b>--strand&nbsp;</b><i>plus|both</i></p>

<p style="margin-left:35%;">When comparing sequences with
the cluster seed, check the <i>plus</i> strand only
(default) or check <i>both</i> strands.
<a name="uc"></a></p>

<p style="margin-left:22%;"><b>--uc</b> <i>filename</i></p>

<p style="margin-left:35%;">Output clustering results in
<i>filename</i> using a tab-separated uclust-like format
with 10 columns and 3 different type of entries (S, H or C).
Each fasta sequence in the input file can be either a
cluster centroid (S) or a hit (H) assigned to a cluster.
Cluster records (C) summarize information (size, centroid
label) for each cluster. In the context of clustering, the
option --uc_allhits has no effect on the --uc output. Column
content varies with the type of entry (S, H or C):</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">1.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Record type: S, H,
or C.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">2.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Cluster number
(zero-based).</p> </td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">3.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Centroid length
(S), query length (H), or cluster size (C).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">4.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Percentage of
similarity with the centroid sequence (H), or set to
&rsquo;*&rsquo; (S, C).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">5.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Match orientation +
or - (H), or set to &rsquo;*&rsquo; (S, C).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">6.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Not used, always
set to &rsquo;*&rsquo; (S, C) or to zero (H).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">7.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Not used, always
set to &rsquo;*&rsquo; (S, C) or to zero (H).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">8.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">set to
&rsquo;*&rsquo; (S, C) or, for H, compact representation of
the pairwise alignment using the CIGAR format (Compact
Idiosyncratic Gapped Alignment Report): M (match/mismatch),
D (deletion) and I (insertion). The equal sign
&rsquo;=&rsquo; indicates that the query is identical to the
centroid sequence.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">9.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Label of the query
sequence (H), or of the centroid sequence (S, C).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">10.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Label of the
centroid sequence (H), or set to &rsquo;*&rsquo; (S, C).</p></td>
</table>

<p style="margin-left:22%;"><a name="unoise_alpha"></a>
<b><br>
--unoise_alpha&nbsp;</b><i>real</i></p>

<p style="margin-left:35%;">Specify the alpha parameter to
the --cluster_unoise command. The default i 2.0.
<a name="usersort"></a></p>

<p style="margin-left:22%;"><b>--usersort</b></p>

<p style="margin-left:35%;">When using --cluster_smallmem,
allow any sequence input order, not just a decreasing length
ordering. <a name="xsize"></a></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>--xsize</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Strip abundance
information from the headers when writing the output
file.</p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>...</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Most searching
options as well as score filtering, gap penalties and
masking also apply to clustering (see the Searching section
for definitions): --alnout, --blast6out, --fastapairs,
--matched, --notmatched, --maxaccept, --maxreject, --samout,
--userout, --userfields</p></td>
</table>

<p style="margin-left:11%; margin-top: 1em">Dereplication
and rereplication options:</p>


<p style="margin-left:22%;"><a name="derep_fulllength"></a>
<b><br>
--derep_fulllength</b> <i>filename</i></p>

<p style="margin-left:35%;">Merge strictly identical
sequences contained in <i>filename</i>. Identical sequences
are defined as having the same length and the same string of
nucleotides (case insensitive, T and U are considered the
same). See the options --sizein and --sizeout to take into
account and compute abundance values. This command does not
support multithreading. <a name="derep_prefix"></a></p>

<p style="margin-left:22%;"><b>--derep_prefix</b>
<i>filename</i></p>

<p style="margin-left:35%;">Merge sequences with identical
prefixes contained in <i>filename</i>. A short sequence
identical to an initial segment (prefix) of another sequence
is considered a replicate of the longer sequence. If a
sequence is identical to the prefix of two or more longer
sequences, it is clustered with the shortest of them. If
they are equally long, it is clustered with the most
abundant. Remaining ties are solved using sequence headers
and sequence input order. Sequence comparisons are case
insensitive, and T and U are considered identical. This
command does not support multithreading.
<a name="maxuniquesize"></a></p>


<p style="margin-left:22%;"><b>--maxuniquesize&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Discard sequences with a
post-dereplication abundance value greater than
<i>integer</i>. <a name="minuniquesize"></a></p>


<p style="margin-left:22%;"><b>--minuniquesize&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Discard sequences with a
post-dereplication abundance value smaller than
<i>integer</i>. <a name="output"></a></p>

<p style="margin-left:22%;"><b>--output</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write the dereplicated
sequences to <i>filename</i>, in fasta format and sorted by
decreasing abundance. Identical sequences receive the header
of the first sequence of their group. If --sizeout is used,
the number of occurrences (i.e. abundance) of each sequence
is indicated at the end of their fasta header using the
pattern</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="22%"></td>
<td width="12%">
</td>
<td width="1%"></td>
<td width="65%">



<p style="margin-top: 1em" valign="top"><a name="relabel"></a>
<b>--relabel</b> <i>string</i> Please see the description of
the same option under Chimera detection for details.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="12%">
</td>
<td width="1%"></td>
<td width="65%">



<p style="margin-top: 1em" valign="top"><a name="relabel_keep"></a>
<b>--relabel_keep</b> When relabelling, keep the old
identifier in the header after a space.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="12%">
</td>
<td width="1%"></td>
<td width="65%">



<p style="margin-top: 1em" valign="top"><a name="relabel_md5"></a>
<b>--relabel_md5</b> Please see the description of the same
option under Chimera detection for details.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="12%">
</td>
<td width="1%"></td>
<td width="65%">



<p style="margin-top: 1em" valign="top"><a name="relabel_sha1"></a>
<b>--relabel_sha1</b> Please see the description of the same
option under Chimera detection for details.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="12%">
</td>
<td width="1%"></td>
<td width="65%">



<p style="margin-top: 1em" valign="top"><a name="rereplicate"></a>
<b>--rereplicate</b> <i>filename</i> Duplicate each sequence
the number of times indicated by the abundance of each
sequence in the specified file (option --sizein is always
implied). The sequence labels are identical for the same
sequence, unless --relabel, --relabel_sha1 or --relabel_md5
is used to create unique labels. Output is written to the
file specified with the --output option, in FASTA format.
The output file does not contain abundance information
unless --sizeout is specified, in which case an abundance of
1 is used. <a name="sizein"></a></p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="12%">



<p style="margin-top: 1em" valign="top"><b>--sizein</b></p> </td>
<td width="1%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Take into account
the abundance annotations present in the input fasta file
(search for the pattern
&rsquo;[&gt;;]size=<i>integer</i>[;]&rsquo; in sequence
headers). That option is active by default when
rereplicating. <a name="sizeout"></a></p></td>
</table>

<p style="margin-left:22%;"><b>--sizeout</b></p>

<p style="margin-left:35%;">Add abundance annotations to
the output fasta file (add the pattern specified, each
unique sequence receives a new abundance value corresponding
to its total abundance (sum of the abundances of its
occurrences). If --sizein is not specified, input abundances
are set to 1, and each unique sequence receives a new
abundance value corresponding to its number of occurrences
in the input file. <a name="strand"></a></p>


<p style="margin-left:22%;"><b>--strand&nbsp;</b><i>plus|both</i></p>

<p style="margin-left:35%;">When searching for strictly
identical sequences, check the <i>plus</i> strand only
(default) or check <i>both</i> strands.
<a name="topn"></a></p>

<p style="margin-left:22%;"><b>--topn&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Output only the top
<i>integer</i> sequences (i.e. the most abundant).
<a name="uc"></a></p>

<p style="margin-left:22%;"><b>--uc</b> <i>filename</i></p>

<p style="margin-left:35%;">Output full-length or
prefix-dereplication results in <i>filename</i> using a
tab-separated uclust-like format with 10 columns and 3
different type of entries (S, H or C). Each fasta sequence
in the input file can be either a cluster centroid (S) or a
hit (H) assigned to a cluster. Cluster records (C) summarize
information (size, centroid label) for each cluster. In the
context of dereplication, the option --uc_allhits has no
effect on the --uc output. Column content varies with the
type of entry (S, H or C):</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">1.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Record type: S, H,
or C.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">2.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Cluster number
(zero-based).</p> </td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">3.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Sequence length (S,
H), or cluster size (C).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">4.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Percentage of
similarity with the centroid sequence (H), or set to
&rsquo;*&rsquo; (S, C).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">5.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Match orientation +
or - (H), or set to &rsquo;*&rsquo; (S, C).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">6.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Not used, always
set to &rsquo;*&rsquo; (S, C) or 0 (H).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">7.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Not used, always
set to &rsquo;*&rsquo; (S, C) or 0 (H).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">8.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Not used, always
set to &rsquo;*&rsquo;.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">9.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Label of the query
sequence (H), or of the centroid sequence (S, C).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">10.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Label of the
centroid sequence (H), or set to &rsquo;*&rsquo; (S, C).</p></td>
</table>


<p style="margin-left:22%; margin-top: 1em"><a name="xsize"></a>
<b><br>
--xsize</b></p>

<p style="margin-left:32%;">Strip abundance information
from the headers when writing the output file.</p>

<p style="margin-left:11%; margin-top: 1em">Extraction
options:</p>

<p style="margin-left:22%; margin-top: 1em">Sequences with
headers matching certain criteria can be extracted from
FASTA and FASTQ files using the --fastx_getseq,
--fastx_getseqs and --fastx_getsubseq commands.</p>

<p style="margin-left:22%; margin-top: 1em">The
--fastx_getseq command requires the header to match a label
specified with the --label option. If the
--label_substr_match option is given, the label may be a
substring located anywhere in the header, otherwise the
entire header must match the label. These matches are not
case-sensitive. The headers in the input file are truncated
at the first space or tab character unless the
--notrunclabels option is given. The matching sequences will
be written to the files specified with the --fastaout and
--fastqout options, in FASTA and FASTQ format, respectively.
Sequences that do not match are written to the files
specified with the --notmatched and --notmatchedfq options,
respectively.</p>

<p style="margin-left:22%; margin-top: 1em">The
--fastx_getsubseq command is similar to the --fastx_getseq
command, but will extract a subsequence of the matching
sequences. The start position is specifed with the
--subseq_start option and the end position is specified with
the --subseq_end option. The positions are 1-based, meaning
that the first symbol of the sequence is at position 1. If
the start or end position option is not specified, the
default is to start at the first position and end at the
last position in the sequence.</p>

<p style="margin-left:22%; margin-top: 1em">The
--fastx_getseqs command is similar to the --fastx_getseq
command but allows more flexibility in specifying the
label(s) to be matched. A single label may be specified
using the --label option as described above. Alternatively,
a file containing a list of labels to be matched may be
specified with the --labels option. The file must be a plain
text file with one label on each line. The --label_word and
--label_words options may be used to specify either a single
word or a file containing a list of words, respectively, to
be matched. Words are defined as character sequences
delimited either by a character that is not alpha-numeric
(A-Z, a-z, or 0-9) or by the beginning or end of the header.
Word matching is case-sensitive. The --label_field option
will limit the matching of words to a certain field in the
header.</p>


<p style="margin-left:22%; margin-top: 1em"><a name="fastaout"></a>
<b><br>
--fastaout</b> <i>filename</i></p>

<p style="margin-left:35%;">Write the extracted sequences
in FASTA format to the file with the given name.
<a name="fastqout"></a></p>

<p style="margin-left:22%;"><b>--fastqout</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write the extracted sequences
in FASTQ format to the file with the given name. This option
is illegal if the input is in FASTA format.
<a name="fastx_getseq"></a></p>

<p style="margin-left:22%;"><b>--fastx_getseq</b>
<i>filename</i></p>

<p style="margin-left:35%;">Extract sequences from the
given FASTA or FASTQ file. Specify a label to match using
the --label option. Output files are specified with the
--fastaout, --fastqout, --notmatched and --notmatchedfq
options. <a name="fastx_getseqs"></a></p>

<p style="margin-left:22%;"><b>--fastx_getseqs</b>
<i>filename</i></p>

<p style="margin-left:35%;">Extract sequences from the
given FASTA or FASTQ file. Specify the label or labels to
match using one of the following options: --label, --labels,
--label_word, or --label_words. Output files are specified
with the --fastaout, --fastqout, --notmatched and
--notmatchedfq options. <a name="fastx_getsubseq"></a></p>

<p style="margin-left:22%;"><b>--fastx_getsubseq</b>
<i>filename</i></p>

<p style="margin-left:35%;">Extract a certain part of some
of the sequences in the given FASTA or FASTQ file. Specify
labels to match using the --label option. Specify the
subsequence range to be extracted with the --subseq_start
and --subseq_end options. Output files are specified with
the --fastaout, --fastqout, --notmatched and --notmatchedfq
options. <a name="label"></a></p>

<p style="margin-left:22%;"><b>--label</b>
<i>string</i></p>

<p style="margin-left:35%;">Specifiy the label to match in
the sequence header. Unless the --label_substr_match option
is given, the label must match the entire header. The
comparison is not case-sensitive.
<a name="label_field"></a></p>

<p style="margin-left:22%;"><b>--label_field</b>
<i>string</i></p>

<p style="margin-left:35%;">Specify a field name to be used
when matching using the --label_word or --label_words
option. The field name is a string like &quot;abc&quot; that
must precede the word to be matched with an equals sign (=)
in between. The field must be delimited by semicolons or the
beginning or end of the header. The following header will
match the label 123 in the field abc:
&quot;seq1;abc=123&quot;.
<a name="label_substr_match"></a></p>


<p style="margin-left:22%;"><b>--label_substr_match</b></p>

<p style="margin-left:35%;">The labels specified with the
--label or the --labels option may match anywhere in the
header if this option is given. Otherwise a label needs to
match the entire header. <a name="label_word"></a></p>

<p style="margin-left:22%;"><b>--label_word</b>
<i>string</i></p>

<p style="margin-left:35%;">Specifiy a word to match in the
sequence header. Words are defined as strings delimited by
either the start or end of the header or by any symbol that
is not a letter (A-Z, a-z) or digit (0-9). The comparison is
case-sensitive. <a name="label_words"></a></p>

<p style="margin-left:22%;"><b>--label_words</b>
<i>filename</i></p>

<p style="margin-left:35%;">Specify a file containing words
to be matched against the sequence headers. The plain text
file must contain one word on each line. Words are defined
as strings delimited by either the start or end of the
header or by any symbol that is not a letter (A-Z, a-z) or
digit (0-9). The comparison is case-sensitive.
<a name="labels"></a></p>

<p style="margin-left:22%;"><b>--labels</b>
<i>filename</i></p>

<p style="margin-left:35%;">Specify a file containing
labels to be matched against the sequence headers. The plain
text file must contain one label on each line. Unless the
--label_substr_match option is given, a label must match the
entire header. The comparison is not case-sensitive.
<a name="notmatched"></a></p>

<p style="margin-left:22%;"><b>--notmatched</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write the sequences that were
not extracted to the file with the given name, in FASTA
format. <a name="notmatchedfq"></a></p>

<p style="margin-left:22%;"><b>--notmatchedfq</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write the sequences that were
not extracted to the file with the given name, in FASTQ
format. This option is illegal if the input is in FASTA
format. <a name="subseq_end"></a></p>


<p style="margin-left:22%;"><b>--subseq_end&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Specifiy the end position in
the sequences when extracting subsequences using the
--fastx_getsubseq command. Positions are 1-based, so the
sequences start at position 1. The default is to end at the
end of the sequence if this option is not specified.
<a name="subseq_start"></a></p>


<p style="margin-left:22%;"><b>--subseq_start&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Specifiy the starting position
in the sequences when extracting subsequences using the
--fastx_getsubseq command. Positions are 1-based, so the
sequences start at position 1. The default is to start at
the beginning of the sequence (position 1), if this option
is not specified.</p>

<p style="margin-left:11%; margin-top: 1em">FASTA/FASTQ
file processing options:</p>

<p style="margin-left:22%; margin-top: 1em">Analyse, trim,
filter, convert or merge sequences in FASTQ files, or
reverse complement sequences in FASTA or FASTQ files. The
--fastq_chars command can be used to analyse FASTQ files to
identify the quality encoding and the range of quality score
values used. To convert between different FASTQ file
variants, use the --fastq_convert command. Statistical
analysis of the quality and length of the sequences in a
FASTQ file may be performed with the --fastq_stats,
--fastq_eestats, and --fastq_eestats2 commands. Sequences
may be trimmed, filtered and converted by the --fastq_filter
or --fastx_filter commands. Paired-end reads can be merged
using the --fastq_mergepairs command. The --fastx_revcomp
command reverse-complements sequences. Finally, the
--sff_convert command can be used to convert SFF files to
FASTQ.</p>


<p style="margin-left:22%; margin-top: 1em"><a name="eeout"></a></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>--eeout</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">When using
--fastq_filter or --fastq_mergepairs, include the number of
expected errors (ee) in the sequence header of FASTQ and
FASTA files. This option is a synonym of the --fastq_eeout
option. Use the --xee option to remove this information from
headers. <a name="eetabbedout"></a></p></td>
</table>

<p style="margin-left:22%;"><b>--eetabbedout</b>
<i>filename</i></p>

<p style="margin-left:35%;">When specified with the
--fastq_mergepairs command, write statistics with expected
errors of each merged read to the given file. The file is a
tab separated file with four columns: The number of errors
expected in the forward read, the number of expected errors
in the reverse read, the number of observed errors in the
forward read, and the number of observed errors in the
reverse read. The observed number of errors are the number
of differences in the overlap region of the merged sequence
relative to each of the reads in the pair.
<a name="fastaout"></a></p>

<p style="margin-left:22%;"><b>--fastaout</b>
<i>filename</i></p>

<p style="margin-left:35%;">When using --fastq_filter,
--fastq_mergepairs or --fastx_filter, write to the given
FASTA-formatted file the sequences passing the filter, or
the merged sequences. <a name="fastaout_rev"></a></p>

<p style="margin-left:22%;"><b>--fastaout_rev</b>
<i>filename</i></p>

<p style="margin-left:35%;">When using --fastq_filter, or
--fastx_filter, write to the given FASTA-formatted file the
reverse reads passing the filter.
<a name="fastaout_notmerged_fwd"></a></p>


<p style="margin-left:22%;"><b>--fastaout_notmerged_fwd</b>
<i>filename</i></p>

<p style="margin-left:35%;">When using --fastq_mergepairs,
write forward reads not merged to the specified FASTA file.
<a name="fastaout_notmerged_rev"></a></p>


<p style="margin-left:22%;"><b>--fastaout_notmerged_rev</b>
<i>filename</i></p>

<p style="margin-left:35%;">When using --fastq_mergepairs,
write reverse reads not merged to the specified FASTA file.
<a name="fastaout_discarded"></a></p>

<p style="margin-left:22%;"><b>--fastaout_discarded</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write sequences that do not
pass the filter of the --fastq_filter or --fastx_filter
command to the given FASTA-formatted file.
<a name="fastaout_discarded_rev"></a></p>


<p style="margin-left:22%;"><b>--fastaout_discarded_rev</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write reverse reads that do not
pass the filter of the --fastq_filter or --fastx_filter
command to the given FASTA-formatted file.
<a name="fastq_allowmergestagger"></a></p>


<p style="margin-left:22%;"><b>--fastq_allowmergestagger</b></p>

<p style="margin-left:35%;">When using --fastq_mergepairs,
allow to merge staggered read pairs. Staggered pairs are
pairs where the 3&rsquo; end of the reverse read has an
overhang to the left of the 5&rsquo; end of the forward
read. This situation can occur when a very short fragment is
sequenced. The 3&rsquo; overhang of the reverse read is not
included in the merged sequence. The opposite option is the
--fastq_nostagger option. The default is to discard
staggered pairs. <a name="fastq_ascii"></a></p>


<p style="margin-left:22%;"><b>--fastq_ascii&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Define the ASCII character
number used as the basis for the FASTQ quality score. The
default is 33, which is used by the Sanger / Illumina 1.8+
FASTQ format (phred+33). The value 64 is used by the Solexa,
Illumina 1.3+ and Illumina 1.5+ formats (phred+64). Only 33
and 64 are valid arguments.
<a name="fastq_asciiout"></a></p>


<p style="margin-left:22%;"><b>--fastq_asciiout&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --fastq_convert or
--sff_convert, define the ASCII character number used as the
basis for the FASTQ quality score when writing FASTQ output
files. The default is 33. Only 33 and 64 are valid
arguments. <a name="fastq_chars"></a></p>

<p style="margin-left:22%;"><b>--fastq_chars</b>
<i>filename</i></p>

<p style="margin-left:35%;">Summarize the composition of
sequence and quality strings contained in the input FASTQ
file. For each of the four DNA letters, --fastq_chars gives
the number of occurrences of the letter, its relative
frequency and the length of the longest run of that letter.
For each character present in the quality strings,
--fastq_chars gives the ASCII value of the character, its
relative frequency, and the number of times a <i>k</i>-mer
of that character appears at the end of quality strings. The
length of the <i>k</i>-mer can be set using --fastq_tail (4
by default). The command --fastq_chars tries to
automatically detect the quality encoding (Solexa, Illumina
1.3+, Illumina 1.5+ or Illumina 1.8+/Sanger) by analyzing
the range of observed quality score values. In case of
success, --fastq_chars suggests values for the --fastq_ascii
(33 or 64), --fastq_qmin and --fastq_qmax options to be used
with the other commands that require a FASTQ input file.
<a name="fastq_convert"></a></p>

<p style="margin-left:22%;"><b>--fastq_convert</b>
<i>filename</i></p>

<p style="margin-left:35%;">Convert between the different
variants of the FASTQ file format. The quality encoding of
the input file must be specified with the --fastq_ascii
option (either 33 or 64, the default is 33), and the output
quality encoding must be specified with the --fastq_asciiout
option (default 33). The minimum and maximum output quality
scores may be limited using the --fastq_qminout and
--fastq_qmaxout options. The output file is specified with
the --fastqout option. <a name="fastq_eeout"></a></p>

<p style="margin-left:22%;"><b>--fastq_eeout</b></p>

<p style="margin-left:35%;">When using --fastq_filter,
--fastx_filter or --fastq_mergepairs, include the number of
expected errors (ee) in the sequence header of FASTQ and
FASTA files. This option is a synonym of the --eeout option.
Use the --xee option to remove this information from
headers. <a name="fastq_eestats"></a></p>

<p style="margin-left:22%;"><b>--fastq_eestats</b>
<i>filename</i></p>

<p style="margin-left:35%;">Analyze a FASTQ file and report
statistics on the distributions of quality scores, error
probabilities and expected accumulated errors. The report, a
table of 21 tab-separated columns, is written to the file
specified with the --output option. The first column
corresponds to the position in the reads (Pos). The second
and third columns correspond to the number of reads (Reads)
and percentage of reads (PctRecs) that include this
position. The remaining columns include information about
the distribution of quality scores in this position (Q),
error probabilities in this position (Pe), and finally the
expected number of accumulated errors from the beginning of
the reads and until the current position (EE). For each of
the Q, Pe and EE distributions, the following statistics are
included: minimum value (Min), lower quartile (Low), median
(Med), mean (Mean), upper quartile (Hi), and maximum value
(Max). The quality encoding and the range of quality values
may be specified with --fastq_ascii --fastq_qmin and
--fastq_qmax. <a name="fastq_eestats2"></a></p>

<p style="margin-left:22%;"><b>--fastq_eestats2</b>
<i>filename</i></p>

<p style="margin-left:35%;">Analyze the specified FASTQ
file and report statistics on the number of sequences that
would be retained at a combination of selected cutoffs for
length truncation and maximum expected errors, that could
potentially be used as arguments to the --fastq_trunclen and
--fastq_maxee options to the --fastq_filter command. The
result, a table of two or more columns, is written to the
file specified with the --output option. There is a line for
each length truncation cutoff. The first column on each line
contains the selected truncation length, while the following
columns contain the number of sequences and, in parenthesis,
the percentage of sequences that would be retained at the
selected EE levels. The truncation length cutoffs may be
specified with the --length_cutoffs option and requires a
list of three comma-separated integers indicating the
shortest cutoff, the longest cutoff, and the increment
between cutoffs. The longest cutoff may be specified with a
star (*) which indicates that the limit is equal to the
longest sequence in the input file. The default setting is
&quot;50,*,50&quot; meaning that truncation lengths of 50,
100, 150 and so on up to the longest sequence length should
be used. The maximum expected error (EE) cutoffs may be
specified with the --ee_cutoffs option which requires a
comma-separated list of floating point numbers as its
argument. The default setting is &quot;0.5,1.0,2.0&quot;
that indicates that expected error levels of 0.5, 1.0 and
2.0 should be used. <a name="fastq_filter"></a></p>

<p style="margin-left:22%;"><b>--fastq_filter</b>
<i>filename</i></p>

<p style="margin-left:35%;">Trim and/or filter sequences in
the given FASTQ file. Similar to the --fastx_filter command,
but works only on FASTQ files. See --fastx_filter for
details. <a name="fastq_join"></a></p>

<p style="margin-left:22%;"><b>--fastq_join</b>
<i>filename</i></p>

<p style="margin-left:35%;">Join paired-end sequence reads
into one sequence and add a gap between them using a padding
sequence. The sequences are not merged as with the
fastq_mergepairs command, but simply joined with a gap. The
forward reads are specified as the argument to this option
and the reverse reads are specified with the --reverse
option. The resulting sequences consist of the forward read,
the padding sequence and the reverse complement of the
reverse read. The padding sequence is specified with the
--join_padgap option and the padding quality is specified
with the --join_padgapq option. The default padding sequence
string is NNNNNNNN and the default padding quality string is
IIIIIIII, corresponding to a base quality score of 40 (a
very high quality score with error probability 0.0001). The
joined sequences are output to the file(s) specified with
the --fastaout or --fastqout options.
<a name="fastq_maxdiffs"></a></p>


<p style="margin-left:22%;"><b>--fastq_maxdiffs&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --fastq_mergepairs,
specify the maximum number of non-matching nucleotides
allowed in the overlap region. That option has a strong
influence on the merging success rate. The default value is
10. <a name="fastq_maxdiffpct"></a></p>


<p style="margin-left:22%;"><b>--fastq_maxdiffpct&nbsp;</b><i>real</i></p>

<p style="margin-left:35%;">When using --fastq_mergepairs,
specify the maximum percentage of non-matching nucleotides
allowed in the overlap region. The default value is 100.0%.
There are other more sophisticated rules in the merging
algorithm that will discard read pairs with a high fraction
of mismatches. <a name="fastq_maxee"></a></p>


<p style="margin-left:22%;"><b>--fastq_maxee&nbsp;</b><i>real</i></p>

<p style="margin-left:35%;">When using --fastq_filter,
--fastq_mergepairs or --fastx_filter, discard sequences with
more than the specified number of expected errors.
<a name="fastq_maxee_rate"></a></p>


<p style="margin-left:22%;"><b>--fastq_maxee_rate&nbsp;</b><i>real</i></p>

<p style="margin-left:35%;">When using --fastq_filter or
--fastx_filter, discard sequences with more than the
specified number of expected errors per base.
<a name="fastq_maxlen"></a></p>


<p style="margin-left:22%;"><b>--fastq_maxlen&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --fastq_filter,
--fastq_mergepairs or --fastx_filter, discard sequences with
more than the specified number of bases.
<a name="fastq_maxmergelen"></a></p>


<p style="margin-left:22%;"><b>--fastq_maxmergelen&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --fastq_mergepairs,
specify the maximum length of the merged sequence. By
default there is no limit. <a name="fastq_maxns"></a></p>


<p style="margin-left:22%;"><b>--fastq_maxns&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --fastq_filter,
--fastq_mergepairs or --fastx_filter, discard sequences with
more than the specified number of N&rsquo;s.
<a name="fastq_mergepairs"></a></p>

<p style="margin-left:22%;"><b>--fastq_mergepairs</b>
<i>filename</i></p>

<p style="margin-left:35%;">Merge paired-end sequence reads
into one sequence. The forward reads are specified as the
argument to this option and the reverse reads are specified
with the --reverse option. The merged sequences are output
to the file(s) specified with the --fastaout or --fastqout
options. The non-merged reads can be output to the files
specified with the --fastaout_notmerged_fwd,
--fastaout_notmerged_rev, --fastqout_notmerged_fwd and
--fastqout_notmerged_rev options. Statistics may be output
to the file specified with the --eetabbedout option.
Sequences are truncated as specified with the
--fastq_truncqual option to remove low-quality bases in the
3&rsquo; end. Sequences shorter than specified with
--fastq_minlen (after truncation) are discarded (1 by
default). Sequences with too many ambiguous bases
(N&rsquo;s), as specified with the --fastq_maxns are also
discarded (no limit by default). Staggered reads are not
merged unless the --fastq_allowmergestagger option is
specified. The minimum length of the overlap region between
the reads may be specified with the --fastq_minovlen option
(default 10). The overlap region may not include more
mismatches than specified with the --fastq_maxdiffs option
(10 by default) or a higher percentage of mismatches than
specified with the --fastq_maxdiffpct option (100.0% by
default), otherwise the read pair is discarded. Additional
rules will avoid merging of reads that cannot be aligned
reliably and unambiguously. The mimimum and maximum length
of the merged sequence may be specified with the
--fastq_minmergelen and --fastq_maxmergelen options,
respectively. Other relevant options are: --fastq_ascii,
--fastq_maxee, --fastq_nostagger, --fastq_qmax,
--fastq_qmaxout, --fastq_qmin, --fastq_qminout, and
--label_suffix. <a name="fastq_minlen"></a></p>


<p style="margin-left:22%;"><b>--fastq_minlen&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --fastq_filter,
--fastq_mergepairs or --fastx_filter, discard sequences with
less than the specified number of bases (default 1).
<a name="fastq_minmergelen"></a></p>


<p style="margin-left:22%;"><b>--fastq_minmergelen&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --fastq_mergepairs,
specify the minimum length of the merged sequence. The
default is 1. <a name="fastq_minovlen"></a></p>


<p style="margin-left:22%;"><b>--fastq_minovlen&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --fastq_mergepairs,
specify the minimum overlap between the merged reads. The
default is 10. <a name="fastq_nostagger"></a></p>

<p style="margin-left:22%;"><b>--fastq_nostagger</b></p>

<p style="margin-left:35%;">When using --fastq_mergepairs,
forbid the merging of staggered read pairs. This is the
default behaviour of --fastq_mergepairs. To change that
behaviour, see the --fastq_allowmergestagger option.
<a name="fastq_qmax"></a></p>


<p style="margin-left:22%;"><b>--fastq_qmax&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Specify the maximum quality
score accepted when reading FASTQ files. The default is 41,
which is usual for recent Sanger/Illumina 1.8+ files.
<a name="fastq_qmaxout"></a></p>


<p style="margin-left:22%;"><b>--fastq_qmaxout&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --fastq_convert or
--sff_convert, specify the maximum quality score used when
writing FASTQ files. The default is 41, which is usual for
recent Sanger/Illumina 1.8+ files. Older formats may use a
maximum quality score of 40. <a name="fastq_qmin"></a></p>


<p style="margin-left:22%;"><b>--fastq_qmin&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Specify the minimum quality
score accepted for FASTQ files. The default is 0, which is
usual for recent Sanger/Illumina 1.8+ files. Older formats
may use scores between -5 and 2.
<a name="fastq_qminout"></a></p>


<p style="margin-left:22%;"><b>--fastq_qminout&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --fastq_convert or
--sff_convert, specify the minimum quality score used when
writing FASTQ files. The default is 0, which is usual for
Sanger/Illumina 1.8+ files. Older versions of the format may
use scores between -5 and 2. <a name="fastq_stats"></a></p>

<p style="margin-left:22%;"><b>--fastq_stats</b>
<i>filename</i></p>

<p style="margin-left:35%;">Analyze a FASTQ file and report
the number of reads it contains. The quality encoding and
the range of quality values may be specified with
--fastq_ascii --fastq_qmin and --fastq_qmax. That command
requires the --log option and outputs the following detailed
statistics on read length, quality score, length vs. quality
distributions, and length / quality filtering: <br>
Read length distribution:</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">1.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">L: read length.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">2.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">N: number of
reads.</p> </td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">3.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Pct: fraction of
reads with this length.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">4:</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">AccPct: fraction of
reads with this length or longer.</p></td>
</table>

<p style="margin-left:35%;">Quality score distribution:</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">1.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">ASCII: character
encoding the quality score.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">2.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Q: Phred quality
score.</p> </td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">3.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Pe: probability of
error associated with the quality score.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">4.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">N: number of bases
with this quality score.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">5.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Pct: fraction of
bases with this quality score.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">6:</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">AccPct: fraction of
bases with this quality score or higher.</p></td>
</table>

<p style="margin-left:35%;">Length vs. quality
distribution:</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">1.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">L: position in
reads (starting from position 2).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">2.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">PctRecs: fraction
of reads with at least this length.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">3.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">AvgQ: average
quality score over all reads up to this position.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">4.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">P(AvgQ): error
probability corresponding to AvgQ.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">5.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">AvgP: average error
probability.</p> </td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">6:</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">AvgEE: average
expected error over all reads up to this position.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">7:</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Rate: growth rate
of AvgEE between this position and position - 1.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">8:</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">RatePct: Rate (as
explained above) expressed as a percentage.</p></td>
</table>

<p style="margin-left:35%;">Effect of expected error and
length filtering:</p>

<p style="margin-left:46%;">The first column indicates read
lengths (<i>L</i>). The next four columns indicate the
number of reads that would be retained by the --fastq_filter
command if the reads were truncated at length <i>L</i>
(option --fastq_trunclen <i>L</i>) and filtered to have a
maximum expected error of 1.0, 0.5, 0.25 or 0.1 (with the
option --fastq_maxee <i>float</i>). The last four columns
indicate the fraction of reads that would be retained by the
--fastq_filter command using the same length and maximum
expected error parameters.</p>

<p style="margin-left:35%;">Effect of minimum quality and
length filtering:</p>

<p style="margin-left:46%;">The first column indicates read
lengths (<i>Len</i>). The next four columns indicate the
fraction of reads that would be retained by the
--fastq_filter command if the reads were truncated at length
<i>Len</i> (option --fastq_trunclen <i>Len</i>) or at the
first position with a quality <i>Q</i> below 5, 10, 15 or 20
(option --fastq_truncqual <i>Q</i>).</p>

<p style="margin-left:22%;"><a name="fastq_stripleft"></a>
<b><br>
--fastq_stripleft&nbsp;</b><i>positive integer</i></p>

<p style="margin-left:35%;">When using --fastq_filter or
--fastx_filter, strip the specified number of bases from the
left end of the reads. <a name="fastq_stripright"></a></p>


<p style="margin-left:22%;"><b>--fastq_stripright&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --fastq_filter or
--fastx_filter, strip the specified number of bases from the
right end of the reads. <a name="fastq_tail"></a></p>


<p style="margin-left:22%;"><b>--fastq_tail&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --fastq_chars, count
the number of times a series of characters of length
<i>k</i> appears at the end of quality strings. By default,
<i>k</i> = 4. <a name="fastq_truncee"></a></p>


<p style="margin-left:22%;"><b>--fastq_truncee&nbsp;</b><i>real</i></p>

<p style="margin-left:35%;">When using --fastq_filter or
--fastx_filter, truncate sequences so that their total
expected error is not higher than the specified value.
<a name="fastq_trunclen"></a></p>


<p style="margin-left:22%;"><b>--fastq_trunclen&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --fastq_filter or
--fastx_filter, truncate sequences to the specified length.
Shorter sequences are discarded.
<a name="fastq_trunclen_keep"></a></p>


<p style="margin-left:22%;"><b>--fastq_trunclen_keep&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --fastq_filter or
--fastx_filter, truncate sequences to the specified length.
Shorter sequences are not discarded.
<a name="fastq_truncqual"></a></p>


<p style="margin-left:22%;"><b>--fastq_truncqual&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --fastq_filter or
--fastx_filter, truncate sequences starting from the first
base with the specified base quality score value or lower.
<a name="fastqout"></a></p>

<p style="margin-left:22%;"><b>--fastqout</b>
<i>filename</i></p>

<p style="margin-left:35%;">When using --fastq_filter,
--fastq_mergepairs or --fastx_filter, write to the given
FASTQ-formatted file the sequences passing the filter, or
the merged sequences. <a name="fastqout_rev"></a></p>

<p style="margin-left:22%;"><b>--fastqout_rev</b>
<i>filename</i></p>

<p style="margin-left:35%;">When using --fastq_filter or
--fastx_filter, write to the given FASTQ-formatted file the
reverse reads passing the filter.
<a name="fastqout_discarded"></a></p>

<p style="margin-left:22%;"><b>--fastqout_discarded</b>
<i>filename</i></p>

<p style="margin-left:35%;">When using --fastq_filter or
--fastx_filter, write sequences that do not pass the filter
to the given FASTQ-formatted file.
<a name="fastqout_discarded_rev"></a></p>


<p style="margin-left:22%;"><b>--fastqout_discarded_rev</b>
<i>filename</i></p>

<p style="margin-left:35%;">When using --fastq_filter or
--fastx_filter, write reverse reads that do not pass the
filter to the given FASTQ-formatted file.
<a name="fastqout_notmerged_fwd"></a></p>


<p style="margin-left:22%;"><b>--fastqout_notmerged_fwd</b>
<i>filename</i></p>

<p style="margin-left:35%;">When using --fastq_mergepairs,
write forward reads not merged to the specified FASTQ file.
<a name="fastqout_notmerged_rev"></a></p>


<p style="margin-left:22%;"><b>--fastqout_notmerged_rev</b>
<i>filename</i></p>

<p style="margin-left:35%;">When using --fastq_mergepairs,
write reverse reads not merged to the specified FASTQ file.
<a name="fastx_filter"></a></p>

<p style="margin-left:22%;"><b>--fastx_filter</b>
<i>filename</i></p>

<p style="margin-left:35%;">Trim and/or filter the
sequences in the given FASTA or FASTQ file and output the
remaining sequences to the FASTQ file specified with the
--fastqout option and/or to the FASTA file specified with
the --fastaout option. Discarded sequences are written to
the files specified with the --fastaout_discarded and
--fastqout_discarded options. The input format (FASTA or
FASTQ) is automatically detected. If the input consists of
paired sequences, an input file with reverse reads may be
specified with the --reverse option, and corresponding
output will be written to the files specified with the
--fastqout_rev, --fastaout_rev, --fastqout_discarded_rev,
and --fastaout_discarded_rev options. Output can not be
written to FASTQ files if the input is in FASTA format. The
sequences are first trimmed and then filtered based on the
remaining bases. Sequences may be trimmed using the options
--fastq_stripleft, --fastq_stripright, --fastq_truncee,
--fastq_trunclen, --fastq_trunclen_keep and
--fastq_truncqual. The sequences may be filtered using the
options --fastq_maxee, --fastq_maxee_rate, --fastq_maxlen,
--fastq_maxns, --fastq_minlen (default 1), --fastq_trunclen,
--maxsize, and --minsize. Sequences not satisfying the
requirements are discarded. For pairs of sequences, both
sequences in a pair must satisfy the requirements, otherwise
both are discarded. If no shortening or filtering options
are given, all sequences are written to the output files,
possibly after conversion from FASTQ to FASTA format. The
--relabel option may be used to relabel the output
sequences. The --eeout option may be used to output the
expected number of errors in each sequence. After all
sequences have been processed, the number of kept and
discarded sequences will be shown, as well as how many of
the kept sequences were trimmed.
<a name="fastx_revcomp"></a></p>

<p style="margin-left:22%;"><b>--fastx_revcomp</b>
<i>filename</i></p>

<p style="margin-left:35%;">Reverse-complement the
sequences in the given FASTA or FASTQ file to a file
specified with the --fastaout and/or --fastqout options. If
the input file is in FASTA format, the output can not be
written back to a FASTQ file due to missing base quality
scores. <a name="join_padgap"></a></p>


<p style="margin-left:22%;"><b>--join_padgap&nbsp;</b><i>string</i></p>

<p style="margin-left:35%;">When running --fastq_join, use
the <i>string</i> as a sequence padding string. The default
is NNNNNNNN (8 N&rsquo;s). <a name="join_padgapq"></a></p>


<p style="margin-left:22%;"><b>--join_padgapq&nbsp;</b><i>string</i></p>

<p style="margin-left:35%;">When running --fastq_join, use
the <i>string</i> as a quality padding string. The default
is a string of I&rsquo;s equal in length to the sequence
padding string. The letter I corresponds to a base quality
score of 40 indicating a very high quality base with error
probability of 0.0001. <a name="label_suffix"></a></p>


<p style="margin-left:22%;"><b>--label_suffix&nbsp;</b><i>string</i></p>

<p style="margin-left:35%;">When using --fastx_revcomp or
--fastq_mergepairs, add the suffix <i>string</i> to sequence
headers. <a name="maxsize"></a></p>


<p style="margin-left:22%;"><b>--maxsize&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --fastq_filter or
--fastx_filter, discard sequences with an abundance higher
than the specified value. <a name="minsize"></a></p>


<p style="margin-left:22%;"><b>--minsize&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --fastq_filter or
--fastx_filter, discard sequences with an abundance lower
than the specified value. <a name="output"></a></p>

<p style="margin-left:22%;"><b>--output</b>
<i>filename</i></p>

<p style="margin-left:35%;">When using --fastq_eestats or
--fastq_eestats2, write tabulated results to
<i>filename</i>. See --fastq_eestats&rsquo;s and
--fastq_eestats2&rsquo;s documentation for a complete
description of the table. <a name="relabel_keep"></a></p>

<p style="margin-left:22%;"><b>--relabel_keep</b></p>

<p style="margin-left:35%;">When using --relabel, keep the
old identifier in the header after a space.
<a name="relabel"></a></p>

<p style="margin-left:22%;"><b>--relabel</b>
<i>string</i></p>

<p style="margin-left:35%;">Please see the description of
the same option under Chimera detection for details.
<a name="relabel_md5"></a></p>

<p style="margin-left:22%;"><b>--relabel_md5</b></p>

<p style="margin-left:35%;">Please see the description of
the same option under Chimera detection for details.
<a name="relabel_sha1"></a></p>

<p style="margin-left:22%;"><b>--relabel_sha1</b></p>

<p style="margin-left:35%;">Please see the description of
the same option under Chimera detection for details.
<a name="reverse"></a></p>

<p style="margin-left:22%;"><b>--reverse</b>
<i>filename</i></p>

<p style="margin-left:35%;">When using --fastq_filter,
--fastx_filter, --fastq_mergepairs or --fastq_join, specify
the FASTQ file containing containing the reverse reads.
<a name="sff_convert"></a></p>

<p style="margin-left:22%;"><b>--sff_convert</b>
<i>filename</i></p>

<p style="margin-left:35%;">Convert the given SFF file to
FASTQ. The FASTQ output file is specified with the
--fastqout option. The sequence may be clipped as specified
in the SFF file if the option --sff_clip is specified,
otherwise no clipping occurs. Bases that would have been
clipped are converted to lower case, while the rest is in
upper case. The output quality encoding may be specified
with the --fastq_asciiout option (default 33). The minimum
and maximum output quality scores may be limited using the
--fastq_qminout and --fastq_qmaxout options.
<a name="sff_clip"></a></p>

<p style="margin-left:22%;"><b>--sff_clip</b></p>

<p style="margin-left:35%;">Specifies that the sequences
converted by the --sff_convert command should be clipped in
both ends as indicated in the SFF file. By default no
clipping is performed. <a name="xsize"></a></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>--xsize</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Strip abundance
information from the headers when writing the output file.
<a name="xee"></a></p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>--xee</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Strip information
about expected errors (ee) from the output file headers.
This information is added by the --fastq_eeout and --eeout
options.</p> </td>
</table>

<p style="margin-left:11%; margin-top: 1em">Masking
options:</p>

<p style="margin-left:22%; margin-top: 1em">An input
sequence can be composed of lower- or uppercase letters.
When soft masking is specified, lower case letters are
treated as symbols that should be masked. Otherwise the case
of the input sequences is ignored.</p>

<p style="margin-left:22%; margin-top: 1em">Masking is
performed by the commands for chimera detection
(uchime_denovo, uchime_ref), clustering (cluster_fast,
cluster_smallmem, cluster_size), masking (maskfasta,
fastx_mask), pairwise alignment (allpairs_global) and
searching (search_exact, usearch_global).</p>

<p style="margin-left:22%; margin-top: 1em">Masking is
usually specified with the --qmask option, while the
--dbmask option is used for the database sequences specified
with the --db option with the --usearch_global,
--search_exact and --uchime_ref commands.</p>

<p style="margin-left:22%; margin-top: 1em">The argument to
the --qmask and --dbmask option may be none, soft or dust.
If the argument is none, the no masking is performed. If the
argument is soft the lower case symbols are masked. Finally,
if the argument is dust, the sequence is masked using the
DUST algorithm by Tatusov and Lipman to mask low-complexity
regions.</p>

<p style="margin-left:22%; margin-top: 1em">If the
--hardmask option is specified, all masked regions are
converted to N&rsquo;s, otherwise masked regions are
indicated by lower case letters.</p>

<p style="margin-left:22%; margin-top: 1em">If any sequence
is masked, the masked version of the sequence (with lower
case letters or N&rsquo;s) is used in all output files.
Otherwise the sequence is unmodified. The exception is the
sequences in the output file specified with the --uchimealns
option, where the input sequences are converted to upper
case first and lower case letters indicate disagreement
between the aligned sequences.</p>

<p style="margin-left:22%; margin-top: 1em">The --qmask
option (or --dbmask for database sequences) may be combined
with the --hardmask option. The results of using the none,
dust or soft argument to --qmask or --dbmask are presented
below, assuming each input sequence contains both lower and
uppercase symbols.</p>

<p style="margin-left:22%; margin-top: 1em">Results if the
--hardmask option is off (default):</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="32%"></td>
<td width="8%">


<p style="margin-top: 1em" valign="top"><b>none:</b></p></td>
<td width="6%"></td>
<td width="54%">


<p style="margin-top: 1em" valign="top">no masking, all
symbols used, no change</p></td>
<tr valign="top" align="left">
<td width="32%"></td>
<td width="8%">


<p style="margin-top: 1em" valign="top"><b>dust:</b></p></td>
<td width="6%"></td>
<td width="54%">


<p style="margin-top: 1em" valign="top">masked symbols
lowercased, rest uppercased</p></td>
<tr valign="top" align="left">
<td width="32%"></td>
<td width="8%">


<p style="margin-top: 1em" valign="top"><b>soft:</b></p></td>
<td width="6%"></td>
<td width="54%">


<p style="margin-top: 1em" valign="top">lowercase symbols
masked, no case changes</p></td>
</table>

<p style="margin-left:22%; margin-top: 1em">Results if the
--hardmask option is on:</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="32%"></td>
<td width="8%">


<p style="margin-top: 1em" valign="top"><b>none:</b></p></td>
<td width="6%"></td>
<td width="54%">


<p style="margin-top: 1em" valign="top">no masking, all
symbols used, no change</p></td>
<tr valign="top" align="left">
<td width="32%"></td>
<td width="8%">


<p style="margin-top: 1em" valign="top"><b>dust:</b></p></td>
<td width="6%"></td>
<td width="54%">


<p style="margin-top: 1em" valign="top">masked symbols
changed to Ns, rest unchanged</p></td>
<tr valign="top" align="left">
<td width="32%"></td>
<td width="8%">


<p style="margin-top: 1em" valign="top"><b>soft:</b></p></td>
<td width="6%"></td>
<td width="54%">


<p style="margin-top: 1em" valign="top">lowercase symbols
masked and changed to Ns</p></td>
</table>

<p style="margin-left:22%; margin-top: 1em">When a sequence
region is masked, words in the region are not included in
the indices used in the heuristic search algorithm. In all
other aspects, the region is treated as other regions.</p>

<p style="margin-left:22%; margin-top: 1em">Regions in
sequences that are hardmasked (with N&rsquo;s) have a zero
alignment score and do not contribute to an alignment.</p>


<p style="margin-left:22%; margin-top: 1em"><a name="fastaout"></a>
<b><br>
--fastaout</b> <i>filename</i></p>

<p style="margin-left:35%;">Write the masked sequences to
<i>filename</i>, in fasta format. Applies only to the
--fastx_mask command. <a name="fastqout"></a></p>

<p style="margin-left:22%;"><b>--fastqout</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write the masked sequences to
<i>filename</i>, in fastq format. Applies only to the
--fastx_mask command. <a name="fastx_mask"></a></p>

<p style="margin-left:22%;"><b>--fastx_mask</b>
<i>filename</i></p>

<p style="margin-left:35%;">Mask regions in sequences
contained in the specified fasta or fastq file. The default
is to mask using DUST (use --qmask to modify that behavior).
The output files are specified with the --fastaout and
--fastqout options. The minimum and maximum percentage of
unmasked residues may be specified with the
--min_unmasked_pct and --max_unmasked_pct options,
respectively. <a name="hardmask"></a></p>

<p style="margin-left:22%;"><b>--hardmask</b></p>

<p style="margin-left:35%;">Symbols in masked regions are
replaced by N&rsquo;s. The default is to replace the masked
regions by lower case letters. <a name="maskfasta"></a></p>

<p style="margin-left:22%;"><b>--maskfasta</b>
<i>filename</i></p>

<p style="margin-left:35%;">Mask regions in sequences
contained in the fasta file <i>filename</i>. The default is
to mask using <i>dust</i> (use --qmask to modify that
behavior). The output file is specified with the --output
option. This command is depreciated, please use --fastx_mask
instead. <a name="max_unmasked_pct"></a></p>

<p style="margin-left:22%;"><b>--max_unmasked_pct</b>
<i>real</i></p>

<p style="margin-left:35%;">Discard sequences with more
than the specified maximum percentage of unmasked residues.
Works only with --fastx_mask.
<a name="min_unmasked_pct"></a></p>

<p style="margin-left:22%;"><b>--min_unmasked_pct</b>
<i>real</i></p>

<p style="margin-left:35%;">Discard sequences with less
than the specified minimum percentage of unmasked residues.
Works only with --fastx_mask. <a name="output"></a></p>

<p style="margin-left:22%;"><b>--output</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write the masked sequences to
<i>filename</i>, in fasta format. Applies only to the
--mask_fasta command. <a name="qmask"></a></p>


<p style="margin-left:22%;"><b>--qmask&nbsp;</b><i>none|dust|soft</i></p>

<p style="margin-left:35%;">If the argument is dust, mask
regions in sequences using the <i>DUST</i> algorithm that
detects simple repeats and low-complexity regions. This is
the default. If the argument is soft, mask the lower case
letters in the input sequence. If the argument is none, do
not mask.</p>

<p style="margin-left:11%; margin-top: 1em">Restriction
site cutting options:</p>

<p style="margin-left:22%; margin-top: 1em">The input
sequences in the file specified with the --cut command are
cut into fragments at all restriction sites matching the
pattern given with the --cut_pattern option. The fragments
on the forward strand are written to the file specified with
the --fastaout file and the fragments on the reverse strand
are written to the file specified with the --fastaout_rev
option. Input sequences that do not match are written to the
file specified with the option --fastaout_discarded, and
their reverse complement are also written to the file
specfied with the --fastaout_discarded_rev option. The
relabel options (--relabel, --relabel_keep, --relabel_md5,
and --relabel_sha1) may be used to relabel the output
sequences). <a name="cut"></a> <b><br>
--cut</b> <i>filename</i></p>

<p style="margin-left:35%;">Specify the input file with
sequences in FASTA format. <a name="cut_pattern"></a></p>

<p style="margin-left:22%;"><b>--cut_pattern</b>
<i>string</i></p>

<p style="margin-left:35%;">Specify the restriction site
cutting pattern and positions. The pattern is a string of
lower- or uppercase letters specifying the nucleotides that
must match, and may include ambiguous nucleotide symbols.
The special characters &quot;^&quot; (circumflex) and
&quot;_&quot; (underscore) are used to indicate the cutting
position on the forward and reverse strand, respectively.
For example, the pattern &quot;G^AATT_C&quot; is the pattern
for the EcoRI restriction site. For such palindromic
patterns (identical to its reverse complement) the command
will output all possible fragments on both strands. For
non-palindromic sites, it may be necessary to run the
command also on the reverse complemented input sequences.
Exactly one cutting site on each strand must be indicated.
<a name="fastaout"></a></p>

<p style="margin-left:22%;"><b>--fastaout</b>
<i>filename</i></p>

<p style="margin-left:35%;">Specify the output file for the
resulting fragments on the forward strand.
<a name="fastaout_rev"></a></p>

<p style="margin-left:22%;"><b>--fastaout_rev</b>
<i>filename</i></p>

<p style="margin-left:35%;">Specify the output file for the
resulting fragments on the reverse strand.
<a name="fastaout_discarded"></a></p>

<p style="margin-left:22%;"><b>--fastaout_discarded</b>
<i>filename</i></p>

<p style="margin-left:35%;">Specify the output file for the
non-matching sequences.
<a name="fastaout_discarded_rev"></a></p>


<p style="margin-left:22%;"><b>--fastaout_discarded_rev</b>
<i>filename</i></p>

<p style="margin-left:35%;">Specify the output file for the
non-matching seqeunces, reverse complemented.</p>

<p style="margin-left:11%; margin-top: 1em">Pairwise
alignment options:</p>

<p style="margin-left:22%; margin-top: 1em">The results of
the n * (n - 1) / 2 pairwise alignments are written to the
result files specified with --alnout, --blast6out,
--fastapairs --matched, --notmatched, --samout, --uc or
--userout (see Searching section below). Specify either the
--acceptall option to output all pairwise alignments, or
specify an identity level with --id to discard weak
alignments. Most other accept/reject options (see Searching
options below) may also be used. Sequences are aligned on
their <i>plus</i> strand only. Masking is performed as usual
and specified with --qmask and --hardmask.
<a name="acceptall"></a> <b><br>
--acceptall</b></p>

<p style="margin-left:35%;">Write the results of all
alignments to output files. This option overrides all other
accept/reject options (including --id).
<a name="allpairs_global"></a></p>

<p style="margin-left:22%;"><b>--allpairs_global</b>
<i>filename</i></p>

<p style="margin-left:35%;">Perform optimal global pairwise
alignments of all vs. all fasta sequences contained in
<i>filename</i>. This command is multi-threaded.
<a name="id"></a></p>

<p style="margin-left:22%;"><b>--id</b> <i>real</i></p>

<p style="margin-left:35%;">Reject the sequence match if
the pairwise identity is lower than <i>real</i> (value
ranging from 0.0 to 1.0 included).
<a name="threads"></a></p>


<p style="margin-left:22%;"><b>--threads&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Number of computation threads
to use (1 to 1024). The number of threads should be lesser
or equal to the number of available CPU cores. The default
is to use all available resources and to launch one thread
per logical core. <a name="uc"></a></p>

<p style="margin-left:22%;"><b>--uc</b> <i>filename</i></p>

<p style="margin-left:35%;">Output pairwise alignment
results in <i>filename</i> using a tab-separated uclust-like
format with 10 columns. Each sequence is compared to all
other sequences, and all hits (--acceptall) or only some
hits (--id <i>float</i>) are reported, with one pairwise
comparison per line:</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">1.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Record type, always
set to &rsquo;H&rsquo;.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">2.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Ordinal number of
the target sequence (based on input order, starting from
zero).</p> </td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">3.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Sequence
length.</p> </td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">4.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Percentage of
similarity with the target sequence.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">5.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Match orientation,
always set to &rsquo;+&rsquo;.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">6.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Not used, always
set to zero.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">7.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Not used, always
set to zero.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">8.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Compact
representation of the pairwise alignment using the CIGAR
format (Compact Idiosyncratic Gapped Alignment Report): M
(match/mismatch), D (deletion) and I (insertion). The equal
sign &rsquo;=&rsquo; indicates that the query is identical
to the centroid sequence.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">9.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Label of the query
sequence.</p> </td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">10.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Label of the target
sequence.</p> </td>
</table>

<p style="margin-left:11%; margin-top: 1em">Searching
options:</p>

<p style="margin-left:22%;"><a name="alnout"></a> <b><br>
--alnout</b> <i>filename</i></p>

<p style="margin-left:35%;">Write pairwise global
alignments to <i>filename</i> using a human-readable format.
Use --rowlen to modify alignment length. Output order may
vary when using multiple threads. <a name="biomout"></a></p>

<p style="margin-left:22%;"><b>--biomout</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write search results to an OTU
table in the biom version 1.0 file format. The query file
contains the samples, while the database file contains the
OTUs. Sample and OTU identifiers are extracted from the
header of these sequences. See the --biomout option in the
Clustering section for further details.
<a name="blast6out"></a></p>

<p style="margin-left:22%;"><b>--blast6out</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write search results to
<i>filename</i> using a blast-like tab-separated format of
twelve fields (listed below), with one line per query-target
matching (or lack of matching if --output_no_hits is used).
Warning, vsearch uses global pairwise alignments, not
blast&rsquo;s seed-and-extend algorithm. Therefore, some
common blast output values (alignment start and end, evalue,
bit score) are reported differently. Output order may vary
when using multiple threads. A similar output can be obtain
with --userout <i>filename</i> and --userfields
query+target+id+alnlen+mism+opens+qlo+qhi+tlo+thi+evalue+bits.
A complete list and description is available in the section
&rsquo;Userfields&rsquo; of this manual.</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">1.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top"><i>query</i>: query
label.</p> </td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">2.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top"><i>target</i>:
target (database sequence) label. The field is set to
&rsquo;*&rsquo; if there is no alignment.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">3.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top"><i>id</i>:
percentage of identity (real value ranging from 0.0 to
100.0). The percentage identity is defined as 100 *
(matching columns) / (alignment length - terminal gaps). See
fields id0 to id4 for other definitions.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">4.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top"><i>alnlen</i>:
length of the query-target alignment (number of columns).
The field is set to 0 if there is no alignment.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">5.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top"><i>mism</i>: number
of mismatches in the alignment (zero or positive integer
value).</p> </td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">6.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top"><i>opens</i>:
number of columns containing a gap opening (zero or positive
integer value).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">7.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top"><i>qlo</i>: first
nucleotide of the query aligned with the target. Always
equal to 1 if there is an alignment, 0 otherwise (see
<i>qilo</i> to ignore initial gaps).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">8.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top"><i>qhi</i>: last
nucleotide of the query aligned with the target. Always
equal to the length of the pairwise alignment, 0 otherwise
(see <i>qihi</i> to ignore terminal gaps).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">9.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top"><i>tlo</i>: first
nucleotide of the target aligned with the query. Always
equal to 1 if there is an alignment, 0 otherwise (see
<i>tilo</i> to ignore initial gaps).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">10.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top"><i>thi</i>: last
nucleotide of the target aligned with the query. Always
equal to the length of the pairwise alignment, 0 otherwise
(see <i>tihi</i> to ignore terminal gaps).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">11.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top"><i>evalue</i>:
expectancy-value (not computed for nucleotide alignments).
Always set to -1.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">12.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top"><i>bits</i>: bit
score (not computed for nucleotide alignments). Always set
to 0.</p></td>
</table>

<p style="margin-left:22%;"><a name="db"></a> <b><br>
--db</b> <i>filename</i></p>

<p style="margin-left:35%;">Compare query sequences
(specified with --usearch_global) to the fasta-formatted
target sequences contained in <i>filename</i>, using global
pairwise alignment. Alternatively, the name of a
preformatted UDB database created using the makeudb_usearch
command (see below) may be specified.
<a name="dbmask"></a></p>


<p style="margin-left:22%;"><b>--dbmask&nbsp;</b><i>none|dust|soft</i></p>

<p style="margin-left:35%;">Mask regions in the target
database sequences using the dust method or the soft method,
or do not mask (none). Warning, when using soft masking
search commands become case sensitive. The default is to
mask using dust. <a name="dbmatched"></a></p>

<p style="margin-left:22%;"><b>--dbmatched</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write database target sequences
matching at least one query sequence to <i>filename</i>, in
fasta format. If the option --sizeout is used, the number of
queries that matched each target sequence is indicated using
the pattern &quot;;size=<i>integer</i>;&quot;.
<a name="dbnotmatched"></a></p>

<p style="margin-left:22%;"><b>--dbnotmatched</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write database target sequences
not matching query sequences to <i>filename</i>, in fasta
format. <a name="fastapairs"></a></p>

<p style="margin-left:22%;"><b>--fastapairs</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write pairwise alignments of
query and target sequences to <i>filename</i>, in fasta
format. <a name="fulldp"></a></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="22%"></td>
<td width="12%">



<p style="margin-top: 1em" valign="top"><b>--fulldp</b></p> </td>
<td width="1%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Dummy option for
compatibility with usearch. To maximize search sensitivity,
<b>vsearch</b> uses a 8-way 16-bit SIMD vectorized full
dynamic programming algorithm (Needleman-Wunsch), whether or
not --fulldp is specified. <a name="gapext"></a></p></td>
</table>

<p style="margin-left:22%;"><b>--gapext</b>
<i>string</i></p>

<p style="margin-left:35%;">Set penalties for a gap
extension. See --gapopen for a complete description of the
penalty declaration system. The default is to initialize the
six gap extending penalties using a penalty of 2 for
extending internal gaps and a penalty of 1 for extending
terminal gaps, in both query and target sequences (i.e.
2I/1E). <a name="gapopen"></a></p>

<p style="margin-left:22%;"><b>--gapopen</b>
<i>string</i></p>

<p style="margin-left:35%;">Set penalties for a gap
opening. A gap opening can occur in six different contexts:
in the query (Q) or in the target (T) sequence, at the left
(L) or right (R) extremity of the sequence, or inside the
sequence (I). Sequence symbols (Q and T) can be combined
with location symbols (L, I, and R), and numerical values to
declare penalties for all possible contexts:
aQL/bQI/cQR/dTL/eTI/fTR, where abcdef are zero or positive
integers, and &rsquo;/&rsquo; is used as a separator. <br>
To simplify declarations, the location symbols (L, I, and R)
can be combined, the symbol (E) can be used to treat both
extremities (L and R) equally, and the symbols Q and T can
be omitted to treat query and target sequences equally. For
instance, the default is to declare a penalty of 20 for
opening internal gaps and a penalty of 2 for opening
terminal gaps (left or right), in both query and target
sequences (i.e. 20I/2E). If only a numerical value is given,
without any sequence or location symbol, then the penalty
applies to all gap openings. To forbid gap-opening, an
infinite penalty value can be declared with the symbol
&rsquo;*&rsquo;. To use <b>vsearch</b> as a semi-global
aligner, a null-penalty can be applied to the left (L) or
right (R) gaps. <b><br>
vsearch</b> always initializes the six gap opening penalties
using the default parameters (20I/2E). The user is then free
to declare only the values he/she wants to modify. The
<i>string</i> is scanned from left to right, accepted
symbols are (0123456789/LIREQT*), and later values override
previous values. <br>
Please note that <b>vsearch</b>, in contrast to usearch,
only allows integer gap penalties. Because the lowest gap
penalties are 0.5 by default in usearch, all default scores
and gap penalties in <b>vsearch</b> have been doubled to
maintain equivalent penalties and to produce identical
alignments. <a name="hardmask"></a></p>

<p style="margin-left:22%;"><b>--hardmask</b></p>

<p style="margin-left:35%;">Mask sequence regions by
replacing them with Ns instead of setting them to lower case
as is the default. For more information, please see the
Masking section. <a name="id"></a></p>

<p style="margin-left:22%;"><b>--id</b> <i>real</i></p>

<p style="margin-left:35%;">Reject the sequence match if
the pairwise identity is lower than <i>real</i> (value
ranging from 0.0 to 1.0 included). The search process sorts
target sequences by decreasing number of <i>k</i>-mers they
have in common with the query sequence, using that
information as a proxy for sequence similarity. That
efficient pre-filtering also prevents pairwise alignments
with weakly matching targets, as there needs to be at least
6 shared <i>k</i>-mers to start the pairwise alignment, and
at least one out of every 16 <i>k</i>-mers from the query
needs to match the target. Consequently, using values lower
than --id 0.5 is not likely to capture more weakly matching
targets. The pairwise identity is by default defined as the
number of (matching columns) / (alignment length - terminal
gaps). That definition can be modified by --iddef.
<a name="iddef"></a></p>


<p style="margin-left:22%;"><b>--iddef&nbsp;</b><i>0|1|2|3|4</i></p>

<p style="margin-left:35%;">Change the pairwise identity
definition used in --id. Values accepted are:</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">0.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">CD-HIT definition:
(matching columns) / (shortest sequence length).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">1.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">edit distance:
(matching columns) / (alignment length).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">2.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">edit distance
excluding terminal gaps (default definition for --id).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">3.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Marine Biological
Lab definition counting each gap opening (internal or
terminal) as a single mismatch, whether or not the gap was
extended: 1.0 - [(mismatches + gap openings)/(longest
sequence length)]</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="3%">


<p style="margin-top: 1em" valign="top">4.</p></td>
<td width="3%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">BLAST definition,
equivalent to --iddef 1 for global pairwise alignments.</p></td>
</table>

<p style="margin-left:35%; margin-top: 1em">The option
--userfields accepts the fields id0 to id4, in addition to
the field id, to report the pairwise identity values
corresponding to the different definitions.</p>

<p style="margin-left:22%;"><a name="idprefix"></a> <b><br>
--idprefix&nbsp;</b><i>positive integer</i></p>

<p style="margin-left:35%;">Reject the sequence match if
the first <i>integer</i> nucleotides of the target do not
match the query. <a name="idsuffix"></a></p>


<p style="margin-left:22%;"><b>--idsuffix&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Reject the sequence match if
the last <i>integer</i> nucleotides of the target do not
match the query. <a name="leftjust"></a></p>

<p style="margin-left:22%;"><b>--leftjust</b></p>

<p style="margin-left:35%;">Reject the sequence match if
the pairwise alignment begins with gaps.
<a name="match"></a></p>


<p style="margin-left:22%;"><b>--match&nbsp;</b><i>integer</i></p>

<p style="margin-left:35%;">Score assigned to a match (i.e.
identical nucleotides) in the pairwise alignment. The
default value is 2. <a name="matched"></a></p>

<p style="margin-left:22%;"><b>--matched</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write query sequences matching
database target sequences to <i>filename</i>, in fasta
format. <a name="maxaccepts"></a></p>


<p style="margin-left:22%;"><b>--maxaccepts&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Maximum number of hits to
accept before stopping the search. The default value is 1.
This option works in pair with --maxrejects. The search
process sorts target sequences by decreasing number of
<i>k</i>-mers they have in common with the query sequence,
using that information as a proxy for sequence similarity.
After pairwise alignments, if the first target sequence
passes the acceptation criteria, it is accepted as best hit
and the search process stops for that query. If --maxaccepts
is set to a higher value, more hits are accepted. If
--maxaccepts and --maxrejects are both set to 0, the
complete database is searched. <a name="maxdiffs"></a></p>


<p style="margin-left:22%;"><b>--maxdiffs&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Reject the sequence match if
the alignment contains at least <i>integer</i>
substitutions, insertions or deletions.
<a name="maxgaps"></a></p>


<p style="margin-left:22%;"><b>--maxgaps&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Reject the sequence match if
the alignment contains at least <i>integer</i> insertions or
deletions. <a name="maxhits"></a></p>


<p style="margin-left:22%;"><b>--maxhits&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Maximum number of hits to show
once the search is terminated (hits are sorted by decreasing
identity). Unlimited by default. That option applies to
--alnout, --blast6out, --fastapairs, --samout, --uc, or
--userout output files. <a name="maxid"></a></p>

<p style="margin-left:22%;"><b>--maxid</b> <i>real</i></p>

<p style="margin-left:35%;">Reject the sequence match if
the percentage of identity between the two sequences is
greater than <i>real</i>. <a name="maxqsize"></a></p>


<p style="margin-left:22%;"><b>--maxqsize&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Reject query sequences with an
abundance greater than <i>integer</i>.
<a name="maxqt"></a></p>

<p style="margin-left:22%;"><b>--maxqt</b> <i>real</i></p>

<p style="margin-left:35%;">Reject if the query/target
sequence length ratio is greater than <i>real</i>.
<a name="maxrejects"></a></p>


<p style="margin-left:22%;"><b>--maxrejects&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Maximum number of non-matching
target sequences to consider before stopping the search. The
default value is 32. This option works in pair with
--maxaccepts. The search process sorts target sequences by
decreasing number of <i>k</i>-mers they have in common with
the query sequence, using that information as a proxy for
sequence similarity. After pairwise alignments, if none of
the first 32 examined target sequences pass the acceptation
criteria, the search process stops for that query (no hit).
If --maxrejects is set to a higher value, more target
sequences are considered. If --maxaccepts and --maxrejects
are both set to 0, the complete database is searched.
<a name="maxsizeratio"></a></p>

<p style="margin-left:22%;"><b>--maxsizeratio</b>
<i>real</i></p>

<p style="margin-left:35%;">Reject if the query/target
abundance ratio is greater than <i>real</i>.
<a name="maxsl"></a></p>

<p style="margin-left:22%;"><b>--maxsl</b> <i>real</i></p>

<p style="margin-left:35%;">Reject if the shorter/longer
sequence length ratio is greater than <i>real</i>.
<a name="maxsubs"></a></p>


<p style="margin-left:22%;"><b>--maxsubs&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Reject the sequence match if
the pairwise alignment contains more than <i>integer</i>
substitutions. <a name="mid"></a></p>

<p style="margin-left:22%;"><b>--mid</b> <i>real</i></p>

<p style="margin-left:35%;">Reject the sequence match if
the percentage of identity is lower than <i>real</i>
(ignoring all gaps, internal and terminal).
<a name="mincols"></a></p>


<p style="margin-left:22%;"><b>--mincols&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Reject the sequence match if
the alignment length is shorter than <i>integer</i>.
<a name="minqt"></a></p>

<p style="margin-left:22%;"><b>--minqt</b> <i>real</i></p>

<p style="margin-left:35%;">Reject if the query/target
sequence length ratio is lower than <i>real</i>.
<a name="minsizeratio"></a></p>

<p style="margin-left:22%;"><b>--minsizeratio</b>
<i>real</i></p>

<p style="margin-left:35%;">Reject if the query/target
abundance ratio is lower than <i>real</i>.
<a name="minsl"></a></p>

<p style="margin-left:22%;"><b>--minsl</b> <i>real</i></p>

<p style="margin-left:35%;">Reject if the shorter/longer
sequence length ratio is lower than <i>real</i>.
<a name="mintsize"></a></p>


<p style="margin-left:22%;"><b>--mintsize&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Reject target sequences with an
abundance lower than <i>integer</i>.
<a name="minwordmatches"></a></p>


<p style="margin-left:22%;"><b>--minwordmatches&nbsp;</b><i>non-negative
integer</i></p>

<p style="margin-left:35%;">Minimum number of word matches
required for a sequence to be considered further. Default
value is 12 for the default word length 8. For word lengths
3-15, the default minimum word matches are 18, 17, 16, 15,
14, 12, 11, 10, 9, 8, 7, 5 and 3, respectively. If the query
sequence has fewer unique words than the number specified,
all words in the query must match. If the argument is 0, no
word matches are required. <a name="mismatch"></a></p>


<p style="margin-left:22%;"><b>--mismatch&nbsp;</b><i>integer</i></p>

<p style="margin-left:35%;">Score assigned to a mismatch
(i.e. different nucleotides) in the pairwise alignment. The
default value is -4. <a name="mothur_shared_out"></a></p>

<p style="margin-left:22%;"><b>--mothur_shared_out</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write search results to an OTU
table in the mothur &rsquo;shared&rsquo; tab-separated plain
text file format. The query file contains the samples, while
the database file contains the OTUs. Sample and OTU
identifiers are extracted from the header of these
sequences. See the --otutabout option in the Clustering
section for further details. <a name="notmatched"></a></p>

<p style="margin-left:22%;"><b>--notmatched</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write query sequences not
matching database target sequences to <i>filename</i>, in
fasta format. <a name="otutabout"></a></p>

<p style="margin-left:22%;"><b>--otutabout</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write search results to an OTU
table in the classic tab-separated plain text format. The
query file contains the samples, while the database file
contains the OTUs. Sample and OTU identifiers are extracted
from the header of these sequences. See the
--mothur_shared_out option in the Clustering section for
further details. <a name="output_no_hits"></a></p>

<p style="margin-left:22%;"><b>--output_no_hits</b></p>

<p style="margin-left:35%;">Write both matching and
non-matching queries to --alnout, --blast6out, --samout or
--userout output files. Non-matching queries are labelled
&rsquo;No hits&rsquo; in --alnout files.
<a name="pattern"></a></p>

<p style="margin-left:22%;"><b>--pattern</b>
<i>string</i></p>

<p style="margin-left:35%;">This option is ignored. It is
provided for compatibility with usearch.
<a name="qmask"></a></p>


<p style="margin-left:22%;"><b>--qmask&nbsp;</b><i>none|dust|soft</i></p>

<p style="margin-left:35%;">Mask regions in the query
sequences using the dust or the soft algorithms, or do not
mask (none). Warning, when using soft masking search
commands become case sensitive. The default is to mask using
<i>dust</i>. <a name="query_cov"></a></p>

<p style="margin-left:22%;"><b>--query_cov</b>
<i>real</i></p>

<p style="margin-left:35%;">Reject if the fraction of the
query aligned to the target sequence is lower than
<i>real</i>. The query coverage is computed as (matches +
mismatches) / query sequence length. Internal or terminal
gaps are not taken into account.
<a name="rightjust"></a></p>

<p style="margin-left:22%;"><b>--rightjust</b></p>

<p style="margin-left:35%;">Reject the sequence match if
the pairwise alignment ends with gaps.
<a name="rowlen"></a></p>


<p style="margin-left:22%;"><b>--rowlen&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Width of alignment lines in
--alnout output. The default value is 64. Set to 0 to
eliminate wrapping. <a name="samheader"></a></p>

<p style="margin-left:22%;"><b>--samheader</b></p>

<p style="margin-left:35%;">Include header lines to the SAM
file when --samout is specified. The header includes lines
starting with @HD, @SQ and @PG, but no @RG lines (see
<a href="https://github.com/samtools/hts-specs">(link)</a>
&lt;https://github.com/samtools/hts-specs&gt;). By default
no header line is written. <a name="samout"></a></p>

<p style="margin-left:22%;"><b>--samout</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write alignment results to
<i>filename</i> using the SAM format (a tab-separated text
file). When using the --samheader option, the SAM file
starts with header lines. Each non-header line is a SAM
record, which represents either a query-target alignment or
the absence of match for a query (output order may vary when
using multiple threads). Each record contains 11 mandatory
fields and optional fields (see
<a href="https://github.com/samtools/hts-specs">(link)</a>
&lt;https://github.com/samtools/hts-specs&gt; for a complete
description of the format):</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">1.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">query sequence
label.</p> </td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">2.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">combination of
bitwise flags. Possible values are: 0 (top hit), 4 (no hit),
16 (reverse-complemented hit), 256 (secondary hit, i.e. all
hits except the top hit).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">3.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">target sequence
label.</p> </td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">4.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">first position of a
target aligned with the query (always 1 for global pairwise
alignments, 0 if there is no match).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">5.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">mapping quality
(ignored, always set to &rsquo;*&rsquo;).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">6.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">CIGAR string (set
to &rsquo;*&rsquo; if there is no match).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">7.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">name of the target
sequence matching with the next read of the query (for mate
reads only, ignored and always set to &rsquo;*&rsquo;).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">8.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">position of the
primary alignment of the next read of the query (for mate
reads only, ignored and always set to 0).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">9.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">target sequence
length (for multi-segment targets, ignored and always set to
0).</p> </td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">10.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">query sequence
(complete, not only the segment aligned to the target as
usearch does).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">11.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">quality string
(ignored, always set to &rsquo;*&rsquo;).</p></td>
</table>

<p style="margin-left:35%;">Optional fields for
query-target matches (number and order of fields may
vary):</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">12.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">AS:i:? alignment
score (i.e. percentage of identity).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">13.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">XN:i:? next best
alignment score (always set to 0).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">14.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">XM:i:? number of
mismatches.</p> </td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">15.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">XO:i:? number of
gap openings (excluding terminal gaps).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">16.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">XG:i:? number of
gap extensions (excluding terminal gaps).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">17.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">NM:i:? edit
distance to the target (sum of XM and XG).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">18.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">MD:Z:? string for
mismatching positions.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">19.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">YT:Z:UU string
representing the alignment type.</p></td>
</table>

<p style="margin-left:22%;"><a name="search_exact"></a>
<b><br>
--search_exact</b> <i>filename</i></p>

<p style="margin-left:35%;">Search for exact full-length
matches to the query sequences contained in <i>filename</i>
in the database of target sequences (--db). Only 100% exact
matches are reported and this command is much faster than
--usearch_global. The --id, --maxaccepts and --maxrejects
options are ignored, but the rest of the searching options
may be specified. <a name="self"></a></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="22%"></td>
<td width="12%">


<p style="margin-top: 1em" valign="top"><b>--self</b></p></td>
<td width="1%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Reject the sequence
match if the query and target labels are identical.
<a name="selfid"></a></p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="12%">



<p style="margin-top: 1em" valign="top"><b>--selfid</b></p> </td>
<td width="1%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Reject the sequence
match if the query and target sequences are strictly
identical. <a name="sizeout"></a></p></td>
</table>

<p style="margin-left:22%;"><b>--sizeout</b></p>

<p style="margin-left:35%;">Add abundance annotations to
the output of the option --dbmatched (using the pattern
&rsquo;;size=<i>integer</i>;&rsquo;), to report the number
of queries that matched each target.
<a name="strand"></a></p>


<p style="margin-left:22%;"><b>--strand&nbsp;</b><i>plus|both</i></p>

<p style="margin-left:35%;">When searching for similar
sequences, check the <i>plus</i> strand only (default) or
check <i>both</i> strands. <a name="target_cov"></a></p>

<p style="margin-left:22%;"><b>--target_cov</b>
<i>real</i></p>

<p style="margin-left:35%;">Reject the sequence match if
the fraction of the target sequence aligned to the query
sequence is lower than <i>real</i>. The target coverage is
computed as (matches + mismatches) / target sequence length.
Internal or terminal gaps are not taken into account.
<a name="top_hits_only"></a></p>

<p style="margin-left:22%;"><b>--top_hits_only</b></p>

<p style="margin-left:35%;">Only the top hits between the
query and database sequence sets are written to the output
specified with the options --alnout, --samout, --userout,
--blast6out, --uc, --fastapairs, --matched or --notmatched
(but not --dbmatched and --dbnotmatched). For each query,
the top hit is the one presenting the highest percentage of
identity (see the --iddef option to change the way identity
is measured). For a given query, if several top hits present
exactly the same percentage of identity, the number of hits
reported is controlled by the --maxaccepts value (1 by
default). <a name="uc"></a></p>

<p style="margin-left:22%;"><b>--uc</b> <i>filename</i></p>

<p style="margin-left:35%;">Output searching results in
<i>filename</i> using a tab-separated uclust-like format
with 10 columns. When using the --search_exact command, the
table layout is the same than with the --allpairs_global.
When using the --usearch_global command, the table present
two different type of entries: hit (H) or no hit (N). Each
query sequence is compared to all other sequences, and the
best hit (--maxaccept 1) or several hits (--maxaccept &gt;
1) are reported (H). Output order may vary when using
multiple threads. Column content varies with the type of
entry (H or N):</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">1.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Record type: H, or
N (&rsquo;hit&rsquo; or &rsquo;no hit&rsquo;).</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">2.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Ordinal number of
the target sequence (based on input order, starting from
zero). Set to &rsquo;*&rsquo; for N.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">3.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Sequence length.
Set to &rsquo;*&rsquo; for N.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">4.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Percentage of
similarity with the target sequence. Set to &rsquo;*&rsquo;
for N.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">5.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Match orientation +
or -. . Set to &rsquo;.&rsquo; for N.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">6.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Not used, always
set to zero for H, or &rsquo;*&rsquo; for N.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">7.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Not used, always
set to zero for H, or &rsquo;*&rsquo; for N.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">8.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Compact
representation of the pairwise alignment using the CIGAR
format (Compact Idiosyncratic Gapped Alignment Report): M
(match/mismatch), D (deletion) and I (insertion). The equal
sign &rsquo;=&rsquo; indicates that the query is identical
to the centroid sequence. Set to &rsquo;*&rsquo; for N.</p></td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">9.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Label of the query
sequence.</p> </td>
<tr valign="top" align="left">
<td width="46%"></td>
<td width="5%">


<p style="margin-top: 1em" valign="top">10.</p></td>
<td width="1%"></td>
<td width="48%">


<p style="margin-top: 1em" valign="top">Label of the target
centroid sequence. Set to &rsquo;*&rsquo; for N.</p></td>
</table>

<p style="margin-left:22%;"><a name="uc_allhits"></a>
<b><br>
--uc_allhits</b></p>

<p style="margin-left:35%;">When using the --uc option,
show all hits, not just the top hit for each query.
<a name="usearch_global"></a></p>

<p style="margin-left:22%;"><b>--usearch_global</b>
<i>filename</i></p>

<p style="margin-left:35%;">Compare target sequences (--db)
to the fasta-formatted query sequences contained in
<i>filename</i>, using global pairwise alignment.
<a name="userfields"></a></p>

<p style="margin-left:22%;"><b>--userfields</b>
<i>string</i></p>

<p style="margin-left:35%;">When using --userout, select
and order the fields written to the output file. Fields are
separated by &rsquo;+&rsquo; (e.g. query+target+id). See the
&rsquo;Userfields&rsquo; section for a complete list of
fields. <a name="userout"></a></p>

<p style="margin-left:22%;"><b>--userout</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write user-defined
tab-separated output to <i>filename</i>. Select the fields
with the option --userfields. Output order may vary when
using multiple threads. If --userfields is empty or not
present, <i>filename</i> is empty.
<a name="weak_id"></a></p>

<p style="margin-left:22%;"><b>--weak_id</b>
<i>real</i></p>

<p style="margin-left:35%;">Show hits with percentage of
identity of at least <i>real</i>, without terminating the
search. A normal search stops as soon as enough hits are
found (as defined by --maxaccepts, --maxrejects, and --id).
As --weak_id reports weak hits that are not deduced from
--maxaccepts, high --id values can be used, hence preserving
both speed and sensitivity. Logically, <i>real</i> must be
smaller than the value indicated by --id.
<a name="wordlength"></a></p>


<p style="margin-left:22%;"><b>--wordlength&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Length of words (i.e.
<i>k</i>-mers) for database indexing. The range of possible
values goes from 3 to 15, but values near 8 or 9 are
generally recommended. Longer words may reduce the
sensitivity/recall for weak similarities, but can increase
precision. On the other hand, shorter words may increase
sensitivity or recall, but may reduce precision. Computation
time generally increases with shorter words and decreases
with longer words, but it increases again for very long
words. Memory requirements for a part of the index increase
with a factor of 4 each time word length increases by one
nucleotide, and this generally becomes significant for long
words (12 or more). The default value is 8.</p>

<p style="margin-left:11%; margin-top: 1em">Shuffling
options:</p>

<p style="margin-left:22%;">Fasta entries in the input file
are outputted in a pseudo-random order.
<a name="output"></a> <b><br>
--output</b> <i>filename</i></p>

<p style="margin-left:35%;">Write the shuffled sequences to
<i>filename</i>, in fasta format.
<a name="randseed"></a></p>


<p style="margin-left:22%;"><b>--randseed&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When shuffling sequence order,
use <i>integer</i> as seed. A given seed always produces the
same output order (useful for replicability). Set to 0 to
use a pseudo-random seed (default behavior).
<a name="relabel"></a></p>

<p style="margin-left:22%;"><b>--relabel</b>
<i>string</i></p>

<p style="margin-left:35%;">Relabel sequences using the
prefix <i>string</i> and a ticker (1, 2, 3, etc.) to
construct the new headers. Use --sizeout to conserve the
abundance annotations. <a name="relabel_keep"></a></p>

<p style="margin-left:22%;"><b>--relabel_keep</b></p>

<p style="margin-left:35%;">When relabelling, keep the old
identifier in the header after a space.
<a name="relabel_md5"></a></p>

<p style="margin-left:22%;"><b>--relabel_md5</b></p>

<p style="margin-left:35%;">Relabel sequences using the MD5
message digest algorithm applied to each sequence. Former
sequence headers are discarded. The sequence is converted to
upper case and U is replaced by T before the digest is
computed. The MD5 digest is a cryptographic hash function
designed to minimize the probability that two different
inputs gives the same output, even for very similar, but
non-identical inputs. Still, there is always a very small,
but non-zero probability that two different inputs give the
same result. The MD5 digest generates a 128-bit (16-byte)
digest that is represented by 16 hexadecimal numbers (using
32 symbols among 0123456789abcdef). Use --sizeout to
conserve the abundance annotations.
<a name="relabel_sha1"></a></p>

<p style="margin-left:22%;"><b>--relabel_sha1</b></p>

<p style="margin-left:35%;">Relabel sequences using the
SHA1 message digest algorithm applied to each sequence. It
is similar to the --relabel_md5 option but uses the SHA1
algorithm instead of the MD5 algorithm. The SHA1 digest
generates a 160-bit (20-byte) result that is represented by
20 hexadecimal numbers (40 symbols). The probability of a
collision (two non-identical sequences having the same
digest) is smaller for the SHA1 algorithm than it is for the
MD5 algorithm. Use --sizeout to conserve the abundance
annotations. <a name="sizeout"></a></p>

<p style="margin-left:22%;"><b>--sizeout</b></p>

<p style="margin-left:35%;">When using --relabel,
--relabel_md5 or --relabel_sha1, preserve and report
abundance annotations to the output fasta file (using the
pattern &rsquo;;size=<i>integer</i>;&rsquo;).
<a name="shuffle"></a></p>

<p style="margin-left:22%;"><b>--shuffle</b>
<i>filename</i></p>

<p style="margin-left:35%;">Pseudo-randomly shuffle the
order of sequences contained in <i>filename</i>.
<a name="topn"></a></p>

<p style="margin-left:22%;"><b>--topn&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Output only the first
<i>integer</i> sequences after pseudo-random reordering.
<a name="xsize"></a></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>--xsize</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Strip abundance
information from the headers when writing the output
file.</p> </td>
</table>

<p style="margin-left:11%; margin-top: 1em">Sorting
options:</p>

<p style="margin-left:22%;">Fasta entries are sorted by
decreasing abundance (--sortbysize) or sequence length
(--sortbylength). To obtain a stable sorting order, ties are
sorted by decreasing abundance and label increasing
alpha-numerical order (--sortbylength), or just by label
increasing alpha-numerical order (--sortbysize). Label
sorting assumes that all sequences have unique labels. The
same applies to the automatic sorting performed during
chimera checking (--uchime_denovo), dereplication
(--derep_fulllength), and clustering (--cluster_fast and
--cluster_size).</p>


<p style="margin-left:22%; margin-top: 1em"><a name="maxsize"></a>
<b><br>
--maxsize&nbsp;</b><i>positive integer</i></p>

<p style="margin-left:35%;">When using --sortbysize,
discard sequences with an abundance value greater than
<i>integer</i>. <a name="minsize"></a></p>


<p style="margin-left:22%;"><b>--minsize&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">When using --sortbysize,
discard sequences with an abundance value smaller than
<i>integer</i>. <a name="output"></a></p>

<p style="margin-left:22%;"><b>--output</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write the sorted sequences to
<i>filename</i>, in fasta format. <a name="relabel"></a></p>

<p style="margin-left:22%;"><b>--relabel</b>
<i>string</i></p>

<p style="margin-left:35%;">Please see the description of
the same option under Chimera detection for details.
<a name="relabel_keep"></a></p>

<p style="margin-left:22%;"><b>--relabel_keep</b></p>

<p style="margin-left:35%;">When relabelling, keep the old
identifier in the header after a space.
<a name="relabel_md5"></a></p>

<p style="margin-left:22%;"><b>--relabel_md5</b></p>

<p style="margin-left:35%;">Please see the description of
the same option under Chimera detection for details.
<a name="relabel_sha1"></a></p>

<p style="margin-left:22%;"><b>--relabel_sha1</b></p>

<p style="margin-left:35%;">Please see the description of
the same option under Chimera detection for details.
<a name="sizeout"></a></p>

<p style="margin-left:22%;"><b>--sizeout</b></p>

<p style="margin-left:35%;">When using --relabel, report
abundance annotations to the output fasta file (using the
pattern &rsquo;;size=<i>integer</i>;&rsquo;).
<a name="sortbylength"></a></p>

<p style="margin-left:22%;"><b>--sortbylength</b>
<i>filename</i></p>

<p style="margin-left:35%;">Sort by decreasing length the
sequences contained in <i>filename</i>. See the general
options --minseqlength and --maxseqlength to eliminate short
and long sequences. <a name="sortbysize"></a></p>

<p style="margin-left:22%;"><b>--sortbysize</b>
<i>filename</i></p>

<p style="margin-left:35%;">Sort by decreasing abundance
the sequences contained in <i>filename</i> (missing
abundance values are assumed to be &rsquo;;size=1&rsquo;).
See the options --minsize and --maxsize to eliminate rare
and dominant sequences. <a name="topn"></a></p>

<p style="margin-left:22%;"><b>--topn&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Output only the top
<i>integer</i> sequences (i.e. the longest or the most
abundant). <a name="xsize"></a></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>--xsize</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Strip abundance
information from the headers when writing the output
file.</p> </td>
</table>

<p style="margin-left:11%; margin-top: 1em">Subsampling
options:</p>

<p style="margin-left:22%;">Subsampling randomly extracts a
certain number or a certain percentage of the sequences in
the input file. If the --sizein option is in effect, the
abundances of the input sequences is taken into account and
the sampling is performed as if the input sequences were
rereplicated, subsampled and dereplicated before being
written to the output file. The extraction is performed as a
random sampling with a uniform distribution among the input
sequences and is performed without replacement. The input
file is specified with the --fastx_subsample option, the
output files are specified with the --fastaout and
--fastqout options and the amount of sequences to be sampled
is specified with the --sample_pct or --sample_size options.
The sequences not sampled may be written to files specified
with the options --fasta_discarded and --fastq_discarded.
The --fastq_ascii, --fastq_qmin and --fastq_qmax options are
also available.</p>


<p style="margin-left:22%; margin-top: 1em"><a name="fastaout"></a>
<b><br>
--fastaout</b> <i>filename</i></p>

<p style="margin-left:35%;">Write the sampled sequences to
<i>filename</i>, in fasta format.
<a name="fastaout_discarded"></a></p>

<p style="margin-left:22%;"><b>--fastaout_discarded</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write the sequences not sampled
to <i>filename</i>, in fasta format.
<a name="fastq_ascii"></a></p>


<p style="margin-left:22%;"><b>--fastq_ascii&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Define the ASCII character
number used as the basis for the FASTQ quality score. The
default is 33, which is used by the Sanger / Illumina 1.8+
FASTQ format (phred+33). The value 64 is used by the Solexa,
Illumina 1.3+ and Illumina 1.5+ formats (phred+64). Only 33
and 64 are valid arguments. <a name="fastq_qmax"></a></p>


<p style="margin-left:22%;"><b>--fastq_qmax&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Specify the maximum quality
score accepted when reading FASTQ files. The default is 41,
which is usual for recent Sanger/Illumina 1.8+ files.
<a name="fastq_qmin"></a></p>


<p style="margin-left:22%;"><b>--fastq_qmin&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Specify the minimum quality
score accepted for FASTQ files. The default is 0, which is
usual for recent Sanger/Illumina 1.8+ files. Older formats
may use scores between -5 and 2. <a name="fastqout"></a></p>

<p style="margin-left:22%;"><b>--fastqout</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write the sampled sequences to
<i>filename</i>, in fastq format. Requires input in fastq
format. <a name="fastqout_discarded"></a></p>

<p style="margin-left:22%;"><b>--fastqout_discarded</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write the sequences not sampled
to <i>filename</i>, in fastq format. Requires input in fastq
format. <a name="fastx_subsample"></a></p>

<p style="margin-left:22%;"><b>--fastx_subsample</b>
<i>filename</i></p>

<p style="margin-left:35%;">Perform subsampling from the
sequences in the specified input file that is in FASTA or
FASTQ format. <a name="randseed"></a></p>


<p style="margin-left:22%;"><b>--randseed&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Use <i>integer</i> as a seed
for the pseudo-random generator. A given seed always
produces the same output, which is useful for replicability.
Set to 0 to use a pseudo-random seed (default behavior).
<a name="relabel"></a></p>

<p style="margin-left:22%;"><b>--relabel</b>
<i>string</i></p>

<p style="margin-left:35%;">Relabel sequences using the
prefix <i>string</i> and a ticker (1, 2, 3, etc.) to
construct the new headers. Use --sizeout to conserve the
abundance annotations. <a name="relabel_keep"></a></p>

<p style="margin-left:22%;"><b>--relabel_keep</b></p>

<p style="margin-left:35%;">When relabelling, keep the old
identifier in the header after a space.
<a name="relabel_md5"></a></p>

<p style="margin-left:22%;"><b>--relabel_md5</b></p>

<p style="margin-left:35%;">Relabel sequences using the MD5
message digest algorithm applied to each sequence. Former
sequence headers are discarded. The sequence is converted to
upper case and U is replaced by T before the digest is
computed. The MD5 digest is a cryptographic hash function
designed to minimize the probability that two different
inputs give the same output, even for very similar, but
non-identical inputs. Still, there is always a very small,
but non-zero probability that two different inputs give the
same result. The MD5 digest generates a 128-bit (16-byte)
digest that is represented by 16 hexadecimal numbers (using
32 symbols among 0123456789abcdef). Use --sizeout to
conserve the abundance annotations.
<a name="relabel_sha1"></a></p>

<p style="margin-left:22%;"><b>--relabel_sha1</b></p>

<p style="margin-left:35%;">Relabel sequences using the
SHA1 message digest algorithm applied to each sequence. It
is similar to the --relabel_md5 option but uses the SHA1
algorithm instead of the MD5 algorithm. The SHA1 digest
generates a 160-bit (20-byte) result that is represented by
20 hexadecimal numbers (40 symbols). The probability of a
collision (two non-identical sequences having the same
digest) is smaller for the SHA1 algorithm than it is for the
MD5 algorithm. Use --sizeout to conserve the abundance
annotations. <a name="sample_pct"></a></p>


<p style="margin-left:22%;"><b>--sample_pct&nbsp;</b><i>real</i></p>

<p style="margin-left:35%;">Subsample the given percentage
of the input sequences. Accepted values range from 0.0 to
100.0. <a name="sample_size"></a></p>


<p style="margin-left:22%;"><b>--sample_size&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Extract the given number of
sequences. <a name="sizein"></a></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="22%"></td>
<td width="12%">



<p style="margin-top: 1em" valign="top"><b>--sizein</b></p> </td>
<td width="1%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Take the abundance
information of the input file into account, otherwise the
abundance of each sequence is considered to be 1.
<a name="sizeout"></a></p> </td>
</table>

<p style="margin-left:22%;"><b>--sizeout</b></p>

<p style="margin-left:35%;">Write abundance information to
the output file. <a name="xsize"></a></p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>--xsize</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Strip abundance
information from the headers when writing the output
file.</p> </td>
</table>

<p style="margin-left:11%; margin-top: 1em">Taxonomic
classification options:</p>

<p style="margin-left:22%;">The vsearch command --sintax
will classify the input sequences according to the Sintax
algorithm as described by Robert Edgar (2016) in SINTAX: a
simple non-Bayesian taxonomy classifier for 16S and ITS
sequences, BioRxiv, 074161. Preprint. doi: 10.1101/074161
<a href="https://doi.org/10.1101/074161">(link)</a></p>

<p style="margin-left:22%; margin-top: 1em">The name of the
fasta file containing the input sequences to be classified
is given as an argument to the --sintax command. The
reference sequence database is specified with the --db
option. The results are written in a tab delimited text file
whose name is specified with the --tabbedout option. The
--sintax_cutoff option may be used to set a minimum level of
bootstrap support for the taxonomic ranks to be
reported.</p>

<p style="margin-left:22%; margin-top: 1em">Multithreading
is supported. Databases in UDB files are supported. The
strand option may be specified.</p>

<p style="margin-left:22%; margin-top: 1em">The reference
database must contain taxonomic information in the header of
each sequence in the form of a string starting with
&quot;;tax=&quot; and followed by a comma-separated list of
up to eight taxonomic identifiers. Each taxonomic identifier
must start with an indication of the rank by one of the
letters d (for domain) k (kingdom), p (phylum), c (class), o
(order), f (family), g (genus), or s (species). The letter
is followed by a colon (:) and the name of that rank. Commas
and semicolons are not allowed in the name of the rank.</p>

<p style="margin-left:22%; margin-top: 1em">Example:
&quot;&gt;X80725_S000004313;tax=d:Bacteria,p:Proteobacteria,c:Gammaproteobacteria,o:Enterobacteriales,f:Enterobacteriaceae,g:Escherichia/Shigella,s:Escherichia_coli&quot;.</p>


<p style="margin-left:22%; margin-top: 1em"><a name="db"></a>
<b><br>
--db</b> <i>filename</i></p>

<p style="margin-left:35%;">Read the reference sequences
from <i>filename</i>, in FASTA, FASTQ or UDB format. These
sequences needs to be annotated with taxonomy.
<a name="sintax_cutoff"></a></p>


<p style="margin-left:22%;"><b>--sintax_cutoff&nbsp;</b><i>real</i></p>

<p style="margin-left:35%;">Specify a minimum level of
bootstrap support for the taxonomic ranks that will be
included in column 4 of the output file. For instance 0.9,
corresponding to 90%. <a name="sintax"></a></p>

<p style="margin-left:22%;"><b>--sintax</b>
<i>filename</i></p>

<p style="margin-left:35%;">Read the input sequences from
<i>filename</i>, in FASTA or FASTQ format.
<a name="tabbedout"></a></p>

<p style="margin-left:22%;"><b>--tabbedout</b>
<i>filename</i></p>

<p style="margin-left:35%;">Write the results to
<i>filename</i>, in a tab-separated text format. Column 1
contains the query label. Column 2 contains the predicted
taxonomy in the same format as for the reference data, with
bootstrap support indicated in parentheses after each rank.
Column 3 contains the strand. If the --sintax_cutoff option
is used, the predicted taxonomy will be repeated in column 4
while omitting the bootstrap values and including only the
ranks with support at or above the threshold.</p>

<p style="margin-left:11%; margin-top: 1em">UDB
options:</p>

<p style="margin-left:22%;">Databases to be used with the
--usearch_global command may be prepared from FASTA files
and stored to a binary UDB formatted file in order to speed
up searching. This may be worthwhile when searching a large
database repeatedly. The sequences are indexed and stored in
a way that can be quickly loaded into memory. The commands
and options below can be used to create and inspect UDB
files. An UDB file may be specified with the --db option
instead of a FASTA formatted file with the --usearch_global
command.</p>


<p style="margin-left:22%; margin-top: 1em"><a name="dbmask"></a>
<b><br>
--dbmask&nbsp;</b><i>none|dust|soft</i></p>

<p style="margin-left:35%;">Specify the sequence masking
method used with the --makeudb_usearch command, either none,
dust or soft. No masking is performed when none is
specified. When dust is specified, the DUST algorithm will
be used for masking low complexity regions (short repeats
and skewed composition). Lower case letters in the input
file will be masked when soft is specified (soft masking).
<a name="hardmask"></a></p>

<p style="margin-left:22%;"><b>--hardmask</b></p>

<p style="margin-left:35%;">Mask sequences by replacing
letters with N for the --makeudb_usearch command. The
default is to use lower case letters (soft masking).
<a name="makeudb_usearch"></a></p>

<p style="margin-left:22%;"><b>--makeudb_usearch</b>
<i>filename</i></p>

<p style="margin-left:35%;">Create an UDB database file
from the FASTA-formatted sequences in the file with the
given <i>filename</i>. The UDB database is written to the
file specified with the --output option.
<a name="output"></a></p>

<p style="margin-left:22%;"><b>--output</b>
<i>filename</i></p>

<p style="margin-left:35%;">Specify the <i>filename</i> of
a FASTA or UDB output file for the --makeudb_usearch or the
--udb2fasta command, respectively.
<a name="udb2fasta"></a></p>

<p style="margin-left:22%;"><b>--udb2fasta</b>
<i>filename</i></p>

<p style="margin-left:35%;">Read the UDB database in the
file with the given <i>filename</i> and output the sequences
in FASTA format in the file specified by the --output
option. <a name="udbinfo"></a></p>

<p style="margin-left:22%;"><b>--udbinfo</b>
<i>filename</i></p>

<p style="margin-left:35%;">Show information about the UDB
database in the file with the given <i>filename</i>.
<a name="udbstats"></a></p>

<p style="margin-left:22%;"><b>--udbstats</b>
<i>filename</i></p>

<p style="margin-left:35%;">Report statistics about the
indexed words in the UDB database in the file with the given
<i>filename</i>. <a name="wordlength"></a></p>


<p style="margin-left:22%;"><b>--wordlength&nbsp;</b><i>positive
integer</i></p>

<p style="margin-left:35%;">Specify the length of the words
to be used when creating the UDB database index using the
--makeudb_usearch command. Valid numbers range from 3 to 15.
The default is 8.</p>

<p style="margin-left:11%; margin-top: 1em">Userfields
(fields accepted by the --userfields option):</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>aln</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Print a string of M
(match), D (delete, i.e. a gap in the query) and I (insert,
i.e. a gap in the target) representing the pairwise
alignment. Empty field if there is no alignment.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>alnlen</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Print the length of
the query-target alignment (number of columns). The field is
set to 0 if there is no alignment.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>bits</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Bit score (not
computed for nucleotide alignments). Always set to 0.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>caln</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Compact
representation of the pairwise alignment using the CIGAR
format (Compact Idiosyncratic Gapped Alignment Report): M
(match/mismatch), D (deletion) and I (insertion). Empty
field if there is no alignment.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>evalue</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">E-value (not
computed for nucleotide alignments). Always set to -1.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>exts</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Number of columns
containing a gap extension (zero or positive integer
value).</p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>gaps</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Number of columns
containing a gap (zero or positive integer value).</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>id</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Percentage of
identity (real value ranging from 0.0 to 100.0). The
percentage identity is defined as 100 * (matching columns) /
(alignment length - terminal gaps).</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>id0</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">CD-HIT definition
of the percentage of identity (real value ranging from 0.0
to 100.0) using the length of the shortest sequence in the
pairwise alignment as denominator: 100 * (matching columns)
/ (shortest sequence length).</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>id1</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">The percentage of
identity (real value ranging from 0.0 to 100.0) is defined
as the edit distance: 100 * (matching columns) / (alignment
length).</p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>id2</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">The percentage of
identity (real value ranging from 0.0 to 100.0) is defined
as the edit distance, excluding terminal gaps. The field id2
is an alias for the field id.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>id3</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Marine Biological
Lab definition of the percentage of identity (real value
ranging from 0.0 to 100.0), counting each gap opening
(internal or terminal) as a single mismatch, whether or not
the gap was extended, and using the length of the longest
sequence in the pairwise alignment as denominator: 100 *
(1.0 - [(mismatches + gaps) / (longest sequence
length)]).</p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>id4</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">BLAST definition of
the percentage of identity (real value ranging from 0.0 to
100.0), equivalent to --iddef 1 in a context of global
pairwise alignment. The field id4 is always equal to the
field id1.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>ids</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Number of matches
in the alignment (zero or positive integer value).</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>mism</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Number of
mismatches in the alignment (zero or positive integer
value).</p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>opens</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Number of columns
containing a gap opening (zero or positive integer
value).</p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>pairs</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Number of columns
containing only nucleotides. That value corresponds to the
length of the alignment minus the gap-containing columns
(zero or positive integer value).</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>pctgaps</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Number of columns
containing gaps expressed as a percentage of the alignment
length (real value ranging from 0.0 to 100.0).</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>pctpv</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Percentage of
positive columns. When working with nucleotide sequences,
this is equivalent to the percentage of matches (real value
ranging from 0.0 to 100.0).</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>pv</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Number of positive
columns. When working with nucleotide sequences, this is
equivalent to the number of matches (zero or positive
integer value).</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>qcov</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Fraction of the
query sequence that is aligned with the target sequence
(real value ranging from 0.0 to 100.0). The query coverage
is computed as 100.0 * (matches + mismatches) / query
sequence length. Internal or terminal gaps are not taken
into account. The field is set to 0.0 if there is no
alignment.</p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>qframe</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Query frame (-3 to
+3). That field only concerns coding sequences and is not
computed by <b>vsearch</b>. Always set to +0.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>qhi</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Last nucleotide of
the query aligned with the target. Always equal to the
length of the pairwise alignment, 0 otherwise (see
<i>qihi</i> to ignore terminal gaps).</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>qihi</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Last nucleotide of
the query aligned with the target (ignoring terminal gaps).
Nucleotide numbering starts from 1. The field is set to 0 if
there is no alignment.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>qilo</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">First nucleotide of
the query aligned with the target (ignoring initial gaps).
Nucleotide numbering starts from 1. The field is set to 0 if
there is no alignment.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>ql</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Query sequence
length (positive integer value). The field is set to 0 if
there is no alignment.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>qlo</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">First nucleotide of
the query aligned with the target. Always equal to 1 if
there is an alignment, 0 otherwise (see <i>qilo</i> to
ignore initial gaps).</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>qrow</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Print the sequence
of the query segment as seen in the pairwise alignment (i.e.
with gap insertions if need be). Empty field if there is no
alignment.</p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>qs</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Query segment
length. Always equal to query sequence length.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>qstrand</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Query strand
orientation (+ or - for nucleotide sequences). Empty field
if there is no alignment.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>query</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Query label.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>raw</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Raw alignment score
(negative, null or positive integer value). The score is the
sum of match rewards minus mismatch penalties, gap openings
and gap extensions. The field is set to 0 if there is no
alignment.</p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>target</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Target label. The
field is set to &rsquo;*&rsquo; if there is no
alignment.</p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>tcov</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Fraction of the
target sequence that is aligned with the query sequence
(real value ranging from 0.0 to 100.0). The target coverage
is computed as 100.0 * (matches + mismatches) / target
sequence length. Internal or terminal gaps are not taken
into account. The field is set to 0.0 if there is no
alignment.</p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>tframe</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Target frame (-3 to
+3). That field only concerns coding sequences and is not
computed by <b>vsearch</b>. Always set to +0.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>thi</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Last nucleotide of
the target aligned with the query. Always equal to the
length of the pairwise alignment, 0 otherwise (see
<i>tihi</i> to ignore terminal gaps).</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>tihi</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Last nucleotide of
the target aligned with the query (ignoring terminal gaps).
Nucleotide numbering starts from 1. The field is set to 0 if
there is no alignment.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>tilo</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">First nucleotide of
the target aligned with the query (ignoring initial gaps).
Nucleotide numbering starts from 1. The field is set to 0 if
there is no alignment.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>tl</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Target sequence
length (positive integer value). The field is set to 0 if
there is no alignment.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>tlo</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">First nucleotide of
the target aligned with the query. Always equal to 1 if
there is an alignment, 0 otherwise (see <i>tilo</i> to
ignore initial gaps).</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>trow</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Print the sequence
of the target segment as seen in the pairwise alignment
(i.e. with gap insertions if need be). Empty field if there
is no alignment.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>ts</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Target segment
length. Always equal to target sequence length. The field is
set to 0 if there is no alignment.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="10%">


<p style="margin-top: 1em" valign="top"><b>tstrand</b></p></td>
<td width="3%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">Target strand
orientation (+ or - for nucleotide sequences). Always set to
&rsquo;+&rsquo;, so reverse strand matches have tstrand
&rsquo;+&rsquo; and qstrand</p></td>
</table>

<a name="DELIBERATE CHANGES"></a>
<h2>DELIBERATE CHANGES</h2>


<p style="margin-left:11%; margin-top: 1em">If you are a
usearch user, our objective is to make you feel at home.
That&rsquo;s why <b>vsearch</b> was designed to behave like
usearch, to some extent. Like any complex software, usearch
is not free from quirks and inconsistencies. We decided not
to reproduce some of them, and for complete transparency, to
document here the deliberate changes we made.</p>

<p style="margin-left:11%; margin-top: 1em">During a search
with usearch, when using the options --blast6out and
--output_no_hits, for queries with no match the number of
fields reported is 13, where it should be 12. This is
corrected in <b>vsearch</b>.</p>

<p style="margin-left:11%; margin-top: 1em">The field raw
of the --userfields option is not informative in usearch.
This is corrected in <b>vsearch</b>.</p>

<p style="margin-left:11%; margin-top: 1em">The fields qlo,
qhi, tlo, thi now have counterparts (qilo, qihi, tilo, tihi)
reporting alignment coordinates ignoring terminal gaps.</p>

<p style="margin-left:11%; margin-top: 1em">In usearch,
when using the option --output_no_hits, queries that receive
no match are reported in --blast6out file, but not in the
alignment output file. This is corrected in
<b>vsearch</b>.</p>

<p style="margin-left:11%; margin-top: 1em"><b>vsearch</b>
introduces a new --cluster_size command that sorts sequences
by decreasing abundance before clustering.</p>

<p style="margin-left:11%; margin-top: 1em"><b>vsearch</b>
reintroduces --iddef alternative pairwise identity
definitions that were removed from usearch.</p>

<p style="margin-left:11%; margin-top: 1em"><b>vsearch</b>
extends the --topn option to sorting commands.</p>

<p style="margin-left:11%; margin-top: 1em"><b>vsearch</b>
extends the --sizein option to dereplication
(--derep_fulllength) and clustering (--cluster_fast).</p>

<p style="margin-left:11%; margin-top: 1em"><b>vsearch</b>
treats T and U as identical nucleotides during
dereplication.</p>

<p style="margin-left:11%; margin-top: 1em"><b>vsearch</b>
sorting is stabilized by using sequence abundances or
sequences labels as secondary or tertiary keys.</p>

<p style="margin-left:11%; margin-top: 1em"><b>vsearch</b>
by default uses the DUST algorithm for masking
low-complexity regions. Masking behavior is also slightly
changed to be more consistent.</p>

<a name="NOVELTIES"></a>
<h2>NOVELTIES</h2>


<p style="margin-left:11%; margin-top: 1em"><b>vsearch</b>
introduces new commands and new options not present in
usearch 7. They are described in the &rsquo;Options&rsquo;
section of this manual. Here is a short list:</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top">uchime2_denovo,
uchime3_denovo, alignwidth, borderline, fasta_score (chimera
checking)</p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top">cluster_size,
cluster_unoise, clusterout_id, clusterout_sort, profile
(clustering)</p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top">fasta_width,
gzip_decompress, bzip2_decompress (general option)</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top">iddef (clustering,
pairwise alignment, searching)</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top">maxuniquesize
(dereplication)</p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top">relabel_md5 and
relabel_sha1 (chimera detection, dereplication, FASTQ
processing, shuffling, sorting)</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top">shuffle
(shuffling)</p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top">fastq_eestats,
fastq_eestats2, fastq_maxlen, fastq_truncee (FASTQ
processing)</p> </td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">



<p style="margin-top: 1em" valign="top">fastaout_discarded,
fastqout_discarded (subsampling)</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top">rereplicate
(dereplication/rereplication)</p> </td>
</table>

<a name="EXAMPLES"></a>
<h2>EXAMPLES</h2>


<p style="margin-left:11%; margin-top: 1em">Align all
sequences in a database with each other and output all
pairwise alignments:</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--allpairs_global <i>database.fas</i> --alnout
<i>results.aln</i> --acceptall</p>

<p style="margin-left:11%; margin-top: 1em">Check for the
presence of chimeras (<i>de novo</i>); parents should be at
least 1.5 times more abundant than chimeras. Output
non-chimeric sequences in fasta format (no wrapping):</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--uchime_denovo <i>queries.fas</i> --abskew 1.5
--nonchimeras <i>results.fas</i> --fasta_width 0</p>

<p style="margin-left:11%; margin-top: 1em">Cluster with a
97% similarity threshold, collect cluster centroids, and
write cluster descriptions using a uclust-like format:</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--cluster_fast <i>queries.fas</i> --id 0.97 --centroids
<i>centroids.fas</i> --uc <i>clusters.uc</i></p>

<p style="margin-left:11%; margin-top: 1em">Dereplicate the
sequences contained in <i>queries.fas</i>, take into account
the abundance information already present, write unwrapped
fasta sequences to <i>queries_unique.fas</i> with the new
abundance information, discard all sequences with an
abundance of 1:</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--derep_fulllength <i>queries.fas</i> --sizein --fasta_width
0 --sizeout --output <i>queries_unique.fas</i>
--minuniquesize 2</p>

<p style="margin-left:11%; margin-top: 1em">Mask simple
repeats and low complexity regions in the input fasta file
with the DUST algorithm (masked regions are lowercased), and
write the results to the output file:</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--maskfasta <i>queries.fas</i> --qmask dust --output
<i>queries_masked.fas</i></p>

<p style="margin-left:11%; margin-top: 1em">Search queries
in a reference database, with a 80%-similarity threshold,
take terminal gaps into account when calculating pairwise
similarities, output pairwise alignments:</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--usearch_global <i>queries.fas</i> --db
<i>references.fas</i> --id 0.8 --iddef 1 --alnout
<i>results.aln</i></p>

<p style="margin-left:11%; margin-top: 1em">Search a
sequence dataset against itself (ignore self hits), get all
matches with at least 60% similarity, and collect results in
a blast-like tab-separated format. Accept an unlimited
number of hits (--maxaccepts 0), and compare each query to
all other sequences, including unlikely candidates
(--maxrejects 0):</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--usearch_global <i>queries.fas</i> --db <i>queries.fas</i>
--self --id 0.6 --blast6out <i>results.blast6</i>
--maxaccepts 0 --maxrejects 0</p>

<p style="margin-left:11%; margin-top: 1em">Shuffle the
input fasta file (change the order of sequences) in a
repeatable fashion (fixed seed), and write unwrapped fasta
sequences to the output file:</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--shuffle <i>queries.fas</i> --output
<i>queries_shuffled.fas</i> --randseed 13 --fasta_width
0</p>

<p style="margin-left:11%; margin-top: 1em">Sort by
decreasing abundance the sequences contained in
<i>queries.fas</i> (using the
&rsquo;size=<i>integer</i>&rsquo; information), relabel the
sequences while preserving the abundance information (with
--sizeout), keep only sequences with an abundance equal to
or greater than 2:</p>

<p style="margin-left:22%; margin-top: 1em"><b>vsearch</b>
--sortbysize <i>queries.fas</i> --output
<i>queries_sorted.fas</i> --relabel sampleA_ --sizeout
--minsize 2</p>

<a name="AUTHORS"></a>
<h2>AUTHORS</h2>


<p style="margin-left:11%; margin-top: 1em">Implementation
by Torbj&oslash;rn Rognes and Tom&aacute;s Flouri,
documentation by Fr&eacute;d&eacute;ric Mah&eacute;.</p>

<a name="CITATION"></a>
<h2>CITATION</h2>


<p style="margin-left:11%; margin-top: 1em">Rognes T,
Flouri T, Nichols B, Quince C, Mah&eacute; F. (2016)
VSEARCH: a versatile open source tool for metagenomics.
<i>PeerJ</i> 4:e2584 doi: 10.7717/peerj.2584
<a href="https://doi.org/10.7717/peerj.2584">(link)</a></p>

<a name="REPORTING BUGS"></a>
<h2>REPORTING BUGS</h2>


<p style="margin-left:11%; margin-top: 1em">Submit
suggestions and bug-reports at
<a href="https://github.com/torognes/vsearch/issues">(link)</a>
&lt;https://github.com/torognes/vsearch/issues&gt;, send a
pull request on
<a href="https://github.com/torognes/vsearch">(link)</a>
&lt;https://github.com/torognes/vsearch&gt;, or compose a
friendly or curmudgeont e-mail to Torbj&oslash;rn Rognes
<a href="mailto:torognes@ifi.uio.no">(link)</a>
&lt;torognes@ifi.uio.no&gt;.</p>

<a name="AVAILABILITY"></a>
<h2>AVAILABILITY</h2>


<p style="margin-left:11%; margin-top: 1em">Source code and
binaries are available at
&lt;https://github.com/torognes/vsearch&gt;.</p>

<a name="COPYRIGHT"></a>
<h2>COPYRIGHT</h2>


<p style="margin-left:11%; margin-top: 1em">Copyright (C)
2014-2018, Torbj&oslash;rn Rognes, Fr&eacute;d&eacute;ric
Mah&eacute; and Tom&aacute;s Flouri</p>

<p style="margin-left:11%; margin-top: 1em">All rights
reserved.</p>

<p style="margin-left:11%; margin-top: 1em">Contact:
Torbj&oslash;rn Rognes &lt;torognes@ifi.uio.no&gt;,
Department of Informatics, University of Oslo, PO Box 1080
Blindern, NO-0316 Oslo, Norway</p>

<p style="margin-left:11%; margin-top: 1em">This software
is dual-licensed and available under a choice of one of two
licenses, either under the terms of the GNU General Public
License version 3 or the BSD 2-Clause License.</p>

<p style="margin-left:11%; margin-top: 1em"><b>GNU General
Public License version 3</b></p>

<p style="margin-left:11%; margin-top: 1em">This program is
free software: you can redistribute it and/or modify it
under the terms of the GNU General Public License as
published by the Free Software Foundation, either version 3
of the License, or (at your option) any later version.</p>

<p style="margin-left:11%; margin-top: 1em">This program is
distributed in the hope that it will be useful, but WITHOUT
ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
GNU General Public License for more details.</p>

<p style="margin-left:11%; margin-top: 1em">You should have
received a copy of the GNU General Public License along with
this program. If not, see
<a href="http://www.gnu.org/licenses/">(link)</a>
&lt;http://www.gnu.org/licenses/&gt;.</p>

<p style="margin-left:11%; margin-top: 1em"><b>The BSD
2-Clause License</b></p>

<p style="margin-left:11%; margin-top: 1em">Redistribution
and use in source and binary forms, with or without
modification, are permitted provided that the following
conditions are met:</p>

<p style="margin-left:11%; margin-top: 1em">1.
Redistributions of source code must retain the above
copyright notice, this list of conditions and the following
disclaimer.</p>

<p style="margin-left:11%; margin-top: 1em">2.
Redistributions in binary form must reproduce the above
copyright notice, this list of conditions and the following
disclaimer in the documentation and/or other materials
provided with the distribution.</p>

<p style="margin-left:11%; margin-top: 1em">THIS SOFTWARE
IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
&quot;AS IS&quot; AND ANY EXPRESS OR IMPLIED WARRANTIES,
INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
DAMAGE.</p>

<p style="margin-left:11%; margin-top: 1em">We would like
to thank the authors of the following projects for making
their source code available:</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top"><b>vsearch</b>
includes code from Google&rsquo;s CityHash project by Geoff
Pike and Jyrki Alakuijala, providing some excellent hash
functions available under a MIT license.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top"><b>vsearch</b>
includes code derived from Tatusov and Lipman&rsquo;s DUST
program that is in the public domain.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top"><b>vsearch</b>
includes public domain code written by Alexander Peslyak for
the MD5 message digest algorithm.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top"><b>vsearch</b>
includes public domain code written by Steve Reid and others
for the SHA1 message digest algorithm.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top"><b>vsearch</b>
binaries may include code from the zlib library, copyright
Jean-Loup Gailly and Mark Adler.</p></td>
<tr valign="top" align="left">
<td width="22%"></td>
<td width="1%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="2%"></td>
<td width="75%">


<p style="margin-top: 1em" valign="top"><b>vsearch</b>
binaries may include code from the bzip2 library, copyright
Julian R. Seward.</p></td>
</table>

<a name="SEE ALSO"></a>
<h2>SEE ALSO</h2>


<p style="margin-left:11%; margin-top: 1em"><b>swipe</b>,
an extremely fast pairwise local (Smith-Waterman) database
search tool by Torbj&oslash;rn Rognes, available at
<a href="https://github.com/torognes/swipe">(link)</a>
&lt;https://github.com/torognes/swipe&gt;.</p>

<p style="margin-left:11%; margin-top: 1em"><b>swarm</b>, a
fast and accurate amplicon clustering method by
Fr&eacute;d&eacute;ric Mah&eacute; and Torbj&oslash;rn
Rognes, available at
<a href="https://github.com/torognes/swarm">(link)</a>
&lt;https://github.com/torognes/swarm&gt;.</p>

<a name="VERSION HISTORY"></a>
<h2>VERSION HISTORY</h2>


<p style="margin-left:11%; margin-top: 1em">New features
and important modifications of <b>vsearch</b> (short lived
or minor bug releases may not be mentioned):</p>

<p style="margin-left:22%;"><b>v1.0.0&nbsp;</b>released
November 28th, 2014</p>

<p style="margin-left:32%;">First public release.</p>

<p style="margin-left:22%;"><b>v1.0.1&nbsp;</b>released
December 1st, 2014</p>

<p style="margin-left:32%;">Bug fixes (sortbysize,
semicolon after size annotation in headers) and minor
changes (labels as secondary sort key for most sorts, treat
T and U as identical for dereplication, only output size in
--dbmatched file if --sizeout specified).</p>

<p style="margin-left:22%;"><b>v1.0.2&nbsp;</b>released
December 6th, 2014</p>

<p style="margin-left:32%;">Bug fixes (ssse3/sse4.1
requirement, memory leak).</p>

<p style="margin-left:22%;"><b>v1.0.3&nbsp;</b>released
December 6th, 2014</p>

<p style="margin-left:32%;">Bug fix (now writes help to
stdout instead of stderr).</p>

<p style="margin-left:22%;"><b>v1.0.4&nbsp;</b>released
December 8th, 2014</p>

<p style="margin-left:32%;">Added --allpairs_global option.
Reduce memory requirements slightly and eliminate memory
leaks.</p>

<p style="margin-left:22%;"><b>v1.0.5&nbsp;</b>released
December 9th, 2014</p>

<p style="margin-left:32%;">Fixes a minor bug with
--allpairs_global and --acceptall options.</p>

<p style="margin-left:22%;"><b>v1.0.6&nbsp;</b>released
December 14th, 2014</p>

<p style="margin-left:32%;">Fixes a memory allocation bug
in chimera detection (--uchime_ref option).</p>

<p style="margin-left:22%;"><b>v1.0.7&nbsp;</b>released
December 19th, 2014</p>

<p style="margin-left:32%;">Fixes a bug in the output from
chimera detection with the --uchimeout option.</p>

<p style="margin-left:22%;"><b>v1.0.8&nbsp;</b>released
January 22nd, 2015</p>

<p style="margin-left:32%;">Introduces several changes and
bug fixes:</p>

<table width="100%" border=0 rules="none" frame="void"
       cellspacing="0" cellpadding="0">
<tr valign="top" align="left">
<td width="32%"></td>
<td width="2%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="1%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">a new linear memory
aligner for alignment of sequences longer than 5,000
nucleotides,</p> </td>
<tr valign="top" align="left">
<td width="32%"></td>
<td width="2%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="1%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">a new
--cluster_size command that sorts sequences by decreasing
abundance before clustering,</p></td>
<tr valign="top" align="left">
<td width="32%"></td>
<td width="2%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="1%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">meaning of
userfields qlo, qhi, tlo, thi changed for compatibility with
usearch,</p> </td>
<tr valign="top" align="left">
<td width="32%"></td>
<td width="2%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="1%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">new userfields
qilo, qihi, tilo, tihi give alignment coordinates ignoring
terminal gaps,</p></td>
<tr valign="top" align="left">
<td width="32%"></td>
<td width="2%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="1%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">in --uc output
files, a perfect alignment is indicated with a
&rsquo;=&rsquo; sign,</p></td>
<tr valign="top" align="left">
<td width="32%"></td>
<td width="2%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="1%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">the option
--cluster_fast now sorts sequences by decreasing length,
then by decreasing abundance and finally by sequence
identifier,</p> </td>
<tr valign="top" align="left">
<td width="32%"></td>
<td width="2%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="1%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">default
--maxseqlength value set to 50,000 nucleotides,</p></td>
<tr valign="top" align="left">
<td width="32%"></td>
<td width="2%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="1%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">fix for bug in
alignment in rare cases,</p></td>
<tr valign="top" align="left">
<td width="32%"></td>
<td width="2%">


<p style="margin-top: 1em" valign="top">-</p></td>
<td width="1%"></td>
<td width="65%">


<p style="margin-top: 1em" valign="top">fix for lack of
detection of under- or overflow in SIMD aligner.</p></td>
</table>

<p style="margin-left:22%;"><b>v1.0.9&nbsp;</b>released
January 22nd, 2015</p>

<p style="margin-left:32%;">Fixes a bug in the function
sorting sequences by decreasing abundance
(--sortbysize).</p>

<p style="margin-left:22%;"><b>v1.0.10&nbsp;</b>released
January 23rd, 2015</p>

<p style="margin-left:32%;">Fixes a bug where the --sizein
option was ignored and always treated as on, affecting
clustering and dereplication commands.</p>

<p style="margin-left:22%;"><b>v1.0.11&nbsp;</b>released
February 5th, 2015</p>

<p style="margin-left:32%;">Introduces the possibility to
output results in SAM format (for clustering, pairwise
alignment and searching).</p>

<p style="margin-left:22%;"><b>v1.0.12&nbsp;</b>released
February 6th, 2015</p>

<p style="margin-left:32%;">Temporarily fixes a problem
with long headers in FASTA files.</p>

<p style="margin-left:22%;"><b>v1.0.13&nbsp;</b>released
February 17th, 2015</p>

<p style="margin-left:32%;">Fix a memory allocation problem
when computing multiple sequence alignments with the
--msaout and --consout options, as well as a memory leak.
Also increased line buffer for reading FASTA files to
4MB.</p>

<p style="margin-left:22%;"><b>v1.0.14&nbsp;</b>released
February 17th, 2015</p>

<p style="margin-left:32%;">Fix a bug where the multiple
alignment and consensus sequence computed after clustering
ignored the strand of the sequences. Also decreased size of
line buffer for reading FASTA files to 1MB again due to
excessive stack memory usage.</p>

<p style="margin-left:22%;"><b>v1.0.15&nbsp;</b>released
February 18th, 2015</p>

<p style="margin-left:32%;">Fix bug in calculation of
identity metric between sequences when using the MBL
definition (--iddef 3).</p>

<p style="margin-left:22%;"><b>v1.0.16&nbsp;</b>released
February 19th, 2015</p>

<p style="margin-left:32%;">Integrated patches from Debian
for increased compatibility with various architectures.</p>

<p style="margin-left:22%;"><b>v1.1.0&nbsp;</b>released
February 20th, 2015</p>

<p style="margin-left:32%;">Added the --quiet option to
suppress all output to stdout and stderr except for warnings
and fatal errors. Added the --log option to write messages
to a log file.</p>

<p style="margin-left:22%;"><b>v1.1.1&nbsp;</b>released
February 20th, 2015</p>

<p style="margin-left:32%;">Added info about --log and
--quiet options to help text.</p>

<p style="margin-left:22%;"><b>v1.1.2&nbsp;</b>released
March 18th, 2015</p>

<p style="margin-left:32%;">Fix bug with large datasets.
Fix format of help info.</p>

<p style="margin-left:22%;"><b>v1.1.3&nbsp;</b>released
March 18th, 2015</p>

<p style="margin-left:32%;">Fix more bugs with large
datasets.</p>


<p style="margin-left:22%;"><b>v1.2.0-1.2.19&nbsp;</b>released
July 6th to September 8th, 2015</p>

<p style="margin-left:32%;">Several new commands and
options added. Bugs fixed. Documentation updated.</p>

<p style="margin-left:22%;"><b>v1.3.0&nbsp;</b>released
September 9th, 2015</p>

<p style="margin-left:32%;">Changed to autotools build
system.</p>

<p style="margin-left:22%;"><b>v1.3.1&nbsp;</b>released
September 14th, 2015</p>

<p style="margin-left:32%;">Several new commands and
options. Bug fixes.</p>

<p style="margin-left:22%;"><b>v1.3.2&nbsp;</b>released
September 15th, 2015</p>

<p style="margin-left:32%;">Fixed memory leaks. Added
&rsquo;-h&rsquo; shortcut for help. Removed extra
&rsquo;v&rsquo; in version number.</p>

<p style="margin-left:22%;"><b>v1.3.3&nbsp;</b>released
September 15th, 2015</p>

<p style="margin-left:32%;">Fixed bug in hexadecimal digits
of MD5 and SHA1 digests. Added --samheader option.</p>

<p style="margin-left:22%;"><b>v1.3.4&nbsp;</b>released
September 16th, 2015</p>

<p style="margin-left:32%;">Fixed compilation problems with
zlib and bzip2lib.</p>

<p style="margin-left:22%;"><b>v1.3.5&nbsp;</b>released
September 17th, 2015</p>

<p style="margin-left:32%;">Minor configuration/makefile
changes to compile to native CPU and simplify makefile.</p>

<p style="margin-left:22%;"><b>v1.4.0&nbsp;</b>released
September 25th, 2015</p>

<p style="margin-left:32%;">Added --sizeorder option.</p>

<p style="margin-left:22%;"><b>v1.4.1&nbsp;</b>released
September 29th, 2015</p>

<p style="margin-left:32%;">Inserted public domain MD5 and
SHA1 code to eliminate dependency on crypto and openssl
libraries and their licensing issues.</p>

<p style="margin-left:22%;"><b>v1.4.2&nbsp;</b>released
October 2nd, 2015</p>

<p style="margin-left:32%;">Dynamic loading of libraries
for reading gzip and bzip2 compressed files if available.
Circumvention of missing gzoffset function in zlib 1.2.3 and
earlier.</p>

<p style="margin-left:22%;"><b>v1.4.3&nbsp;</b>released
October 3rd, 2015</p>

<p style="margin-left:32%;">Fix a bug with determining
amount of memory on some versions of Apple OS X.</p>

<p style="margin-left:22%;"><b>v1.4.4&nbsp;</b>released
October 3rd, 2015</p>

<p style="margin-left:32%;">Remove debug message.</p>

<p style="margin-left:22%;"><b>v1.4.5&nbsp;</b>released
October 6th, 2015</p>

<p style="margin-left:32%;">Fix memory allocation bug when
reading long FASTA sequences.</p>

<p style="margin-left:22%;"><b>v1.4.6&nbsp;</b>released
October 6th, 2015</p>

<p style="margin-left:32%;">Fix subtle bug in SIMD
alignment code that reduced accuracy.</p>

<p style="margin-left:22%;"><b>v1.4.7&nbsp;</b>released
October 7th, 2015</p>

<p style="margin-left:32%;">Fixes a problem with searching
for or clustering sequences with repeats. In this new
version, vsearch looks at all words occurring at least once
in the sequences in the initial step. Previously only words
occurring exactly once were considered. In addition, vsearch
now requires at least 10 words to be shared by the
sequences, previously only 6 were required. If the query
contains less than 10 words, all words must be present for a
match. This change seems to lead to slightly reduced recall,
but somewhat increased precision, ending up with slightly
improved overall accuracy.</p>

<p style="margin-left:22%;"><b>v1.5.0&nbsp;</b>released
October 7th, 2015</p>

<p style="margin-left:32%;">This version introduces the new
option --minwordmatches that allows the user to specify the
minimum number of matching unique words before a sequence is
considered further. New default values for different word
lengths are also set. The minimum word length is increased
to 7.</p>

<p style="margin-left:22%;"><b>v1.6.0&nbsp;</b>released
October 9th, 2015</p>

<p style="margin-left:32%;">This version adds the
relabeling options (--relabel, --relabel_md5 and
--relabel_sha1) to the shuffle command. It also adds the
--xsize option to the clustering, dereplication, shuffling
and sorting commands.</p>

<p style="margin-left:22%;"><b>v1.6.1&nbsp;</b>released
October 14th, 2015</p>

<p style="margin-left:32%;">Fix bugs and update manual and
help text regarding relabelling. Add all relabelling options
to the subsampling command. Add the --xsize option to
chimera detection, dereplication and fastq filtering
commands. Refactoring of code.</p>

<p style="margin-left:22%;"><b>v1.7.0&nbsp;</b>released
October 14th, 2015</p>

<p style="margin-left:32%;">Add --relabel_keep option.</p>

<p style="margin-left:22%;"><b>v1.8.0&nbsp;</b>released
October 19th, 2015</p>

<p style="margin-left:32%;">Added --search_exact,
--fastx_mask and --fastq_convert commands. Changed most
commands to read FASTQ input files as well as FASTA files.
Modified --fastx_revcomp and --fastx_subsample to write
FASTQ files.</p>

<p style="margin-left:22%;"><b>v1.8.1&nbsp;</b>released
November 2nd, 2015</p>

<p style="margin-left:32%;">Fixes for compatibility with
QIIME and older OS X versions.</p>

<p style="margin-left:22%;"><b>v1.9.0&nbsp;</b>released
November 12th, 2015</p>

<p style="margin-left:32%;">Added the --fastq_mergepairs
command and associated options. This command has not been
tested well yet. Included additional files to avoid
dependency of autoconf for compilation. Fixed an error where
identifiers in fasta headers where not truncated at tabs,
just spaces. Fixed a bug in detection of the file format
(FASTA/FASTQ) of a gzip compressed input file.</p>

<p style="margin-left:22%;"><b>v1.9.1&nbsp;</b>released
November 13th, 2015</p>

<p style="margin-left:32%;">Fixed memory leak and a bug in
score computation in --fastq_mergepairs, and improved
speed.</p>

<p style="margin-left:22%;"><b>v1.9.2&nbsp;</b>released
November 17th, 2015</p>

<p style="margin-left:32%;">Fixed a bug in the computation
of some values with --fastq_stats.</p>

<p style="margin-left:22%;"><b>v1.9.3&nbsp;</b>released
November 19th, 2015</p>

<p style="margin-left:32%;">Workaround for missing
x86intrin.h with old compilers.</p>

<p style="margin-left:22%;"><b>v1.9.4&nbsp;</b>released
December 3rd, 2015</p>

<p style="margin-left:32%;">Fixed incrementation of counter
when relabeling dereplicated sequences.</p>

<p style="margin-left:22%;"><b>v1.9.5&nbsp;</b>released
December 3rd, 2015</p>

<p style="margin-left:32%;">Fixed bug resulting in inferior
chimera detection performance.</p>

<p style="margin-left:22%;"><b>v1.9.6&nbsp;</b>released
January 8th, 2016</p>

<p style="margin-left:32%;">Fixed bug in aligned sequences
produced with --fastapairs and --userout (qrow, trow)
options.</p>

<p style="margin-left:22%;"><b>v1.9.7&nbsp;</b>released
January 12th, 2016</p>

<p style="margin-left:32%;">Masking behavior is changed
somewhat to keep the letter case of the input sequences
unchanged when no masking is performed. Masking is now
performed also during chimera detection. Documentation
updated.</p>

<p style="margin-left:22%;"><b>v1.9.8&nbsp;</b>released
January 22nd, 2016</p>

<p style="margin-left:32%;">Fixed bug causing segfault when
chimera detection is performed on extremely short
sequences.</p>

<p style="margin-left:22%;"><b>v1.9.9&nbsp;</b>released
January 22nd, 2016</p>

<p style="margin-left:32%;">Adjusted default minimum number
of word matches during searches for improved
performance.</p>

<p style="margin-left:22%;"><b>v1.9.10&nbsp;</b>released
January 25th, 2016</p>

<p style="margin-left:32%;">Fixed bug related to masking
and lower case database sequences.</p>

<p style="margin-left:22%;"><b>v1.10.0&nbsp;</b>released
February 11th, 2016</p>

<p style="margin-left:32%;">Parallelized and improved
merging of paired-end reads and adjusted some defaults.
Removed progress indicator when stderr is not a terminal.
Added --fasta_score option to report chimera scores in FASTA
files. Added --rereplicate and --fastq_eestats commands.
Fixed typos. Added relabelling to files produced with
--consout and --profile options.</p>

<p style="margin-left:22%;"><b>v1.10.1&nbsp;</b>released
February 23rd, 2016</p>

<p style="margin-left:32%;">Fixed a bug affecting the
--fastq_mergepairs command causing FASTQ headers to be
truncated at first space (despite the bug fix release 1.9.0
of November 12th, 2015). Full headers are now included in
the output (no matter if --notrunclabels is in effect or
not).</p>

<p style="margin-left:22%;"><b>v1.10.2&nbsp;</b>released
March 18th, 2016</p>

<p style="margin-left:32%;">Fixed a bug causing a
segmentation fault when running --usearch_global with an
empty query sequence. Also fixed a bug causing imperfect
alignments to be reported with an alignment string of
&rsquo;=&rsquo; in uc output files. Fixed typos in man file.
Fixed fasta/fastq processing code regarding presence or
absence of compression library header files.</p>

<p style="margin-left:22%;"><b>v1.11.1&nbsp;</b>released
April 13th, 2016</p>

<p style="margin-left:32%;">Added strand information in UC
file for --derep_fulllength and --derep_prefix. Added
expected errors (ee) to header of FASTA files specified with
--fastaout and --fastaout_discarded when --eeout or
--fastq_eeout option is in effect for fastq_filter and
fastq_mergepairs. The options --eeout and --fastq_eeout are
now equivalent.</p>

<p style="margin-left:22%;"><b>v1.11.2&nbsp;</b>released
June 21st, 2016</p>

<p style="margin-left:32%;">Two bugs were fixed. The first
issue was related to the --query_cov option that used a
different coverage definition than the qcov userfield. The
coverage is now defined as the fraction of the whole query
sequence length that is aligned with matching or mismatching
residues in the target. All gaps are ignored. The other
issue was related to the consensus sequences produced during
clustering when only N&rsquo;s were present in some
positions. Previously these would be converted to A&rsquo;s
in the consensus. The behaviour is changed so that N&rsquo;s
are produced in the consensus, and it should now be more
compatible with usearch.</p>

<p style="margin-left:22%;"><b>v2.0.0&nbsp;</b>released
June 24th, 2016</p>

<p style="margin-left:32%;">This major new version supports
reading from pipes. Two new options are added:
--gzip_decompress and --bzip2_decompress. One of these
options must be specified if reading compressed input from a
pipe, but are not required when reading from ordinary files.
The vsearch header that was previously written to stdout is
now written to stderr. This enables piping of results for
further processing. The file name &rsquo;-&rsquo; now
represent standard input (/dev/stdin) or standard output
(/dev/stdout) when reading or writing files, respectively.
Code for reading FASTA and FASTQ files has been
refactored.</p>

<p style="margin-left:22%;"><b>v2.0.1&nbsp;</b>released
June 30th, 2016</p>

<p style="margin-left:32%;">Avoid segmentation fault when
masking very long sequences.</p>

<p style="margin-left:22%;"><b>v2.0.2&nbsp;</b>released
July 5th, 2016</p>

<p style="margin-left:32%;">Avoid warnings when compiling
with GCC 6.</p>

<p style="margin-left:22%;"><b>v2.0.3&nbsp;</b>released
August 2nd, 2016</p>

<p style="margin-left:32%;">Fixed bad compiler options
resulting in Illegal instruction errors when running
precompiled binaries.</p>

<p style="margin-left:22%;"><b>v2.0.4&nbsp;</b>released
September 1st, 2016</p>

<p style="margin-left:32%;">Improved error message for bad
FASTQ quality values. Improved manual.</p>

<p style="margin-left:22%;"><b>v2.0.5&nbsp;</b>released
September 9th, 2016</p>

<p style="margin-left:32%;">Add options
--fastaout_discarded and --fastqout_discarded to output
discarded sequences from subsampling to separate files.
Updated manual.</p>

<p style="margin-left:22%;"><b>v2.1.0&nbsp;</b>released
September 16th, 2016</p>

<p style="margin-left:32%;">New command: --fastx_filter.
New options: --fastq_maxlen, --fastq_truncee. Allow
--minwordmatches down to 3.</p>

<p style="margin-left:22%;"><b>v2.1.1&nbsp;</b>released
September 23rd, 2016</p>

<p style="margin-left:32%;">Fixed bugs in output to
UC-files. Improved help text and manual.</p>

<p style="margin-left:22%;"><b>v2.1.2&nbsp;</b>released
September 28th, 2016</p>

<p style="margin-left:32%;">Fixed incorrect abundance
output from fastx_filter and fastq_filter when
relabelling.</p>

<p style="margin-left:22%;"><b>v2.2.0&nbsp;</b>released
October 7th, 2016</p>

<p style="margin-left:32%;">Added OTU table generation
options --biomout, --mothur_shared_out and --otutabout to
the clustering and searching commands.</p>

<p style="margin-left:22%;"><b>v2.3.0&nbsp;</b>released
October 10th, 2016</p>

<p style="margin-left:32%;">Allowed zero-length sequences
in FASTA and FASTQ files. Added --fastq_trunclen_keep
option. Fixed bug with output of OTU tables to pipes.</p>

<p style="margin-left:22%;"><b>v2.3.1&nbsp;</b>released
November 16th, 2016</p>

<p style="margin-left:32%;">Fixed bug where
--minwordmatches 0 was interpreted as the default minimum
word matches for the given word length instead of zero. When
used in combination with --maxaccepts 0 and --maxrejects 0
it will allow complete bypass of kmer-based heuristics.</p>

<p style="margin-left:22%;"><b>v2.3.2&nbsp;</b>released
November 18th, 2016</p>

<p style="margin-left:32%;">Fixed bug where vsearch
reported the ordinal number of the target sequence instead
of the cluster number in column 2 on H-lines in the uc
output file after clustering. For search and alignment
commands both usearch and vsearch reports the target
sequence number here.</p>

<p style="margin-left:22%;"><b>v2.3.3&nbsp;</b>released
December 5th, 2016</p>

<p style="margin-left:32%;">A minor speed improvement.</p>

<p style="margin-left:22%;"><b>v2.3.4&nbsp;</b>released
December 9th, 2016</p>

<p style="margin-left:32%;">Fixed bug in output of sequence
profiles and updated documentation.</p>

<p style="margin-left:22%;"><b>v2.4.0&nbsp;</b>released
February 8th, 2017</p>

<p style="margin-left:32%;">Added support for Linux on
Power8 systems (ppc64le) and Windows on x86_64. Improved
detection of pipes when reading FASTA and FASTQ files.
Corrected option for specifiying output from fastq_eestats
command in help text.</p>

<p style="margin-left:22%;"><b>v2.4.1&nbsp;</b>released
March 1st, 2017</p>

<p style="margin-left:32%;">Fixed an overflow bug in
fastq_stats and fastq_eestats affecting analysis of very
large FASTQ files. Fixed maximum memory usage reporting on
Windows.</p>

<p style="margin-left:22%;"><b>v2.4.2&nbsp;</b>released
March 10th, 2017</p>

<p style="margin-left:32%;">Default value for
fastq_minovlen increased to 16 in accordance with help text
and for compatibility with usearch. Minor changes for
improved accuracy of paired-end read merging.</p>

<p style="margin-left:22%;"><b>v2.4.3&nbsp;</b>released
April 6th, 2017</p>

<p style="margin-left:32%;">Fixed bug with progress bar for
shuffling. Fixed missing N-lines in UC files with
usearch_global, search_exact and allpairs_global when the
output_no_hits option was not specified.</p>

<p style="margin-left:22%;"><b>v2.4.4&nbsp;</b>released
August 28th, 2017</p>

<p style="margin-left:32%;">Fixed a few minor bugs,
improved error messages and updated documentation.</p>

<p style="margin-left:22%;"><b>v2.5.0&nbsp;</b>released
October 5th, 2017</p>

<p style="margin-left:32%;">Support for UDB database files.
New commands: fastq_stripright, fastq_eestats2,
makeudb_usearch, udb2fasta, udbinfo, and udbstats. New
general option: no_progress. New options minsize and maxsize
to fastx_filter. Minor bug fixes, error message improvements
and documentation updates.</p>

<p style="margin-left:22%;"><b>v2.5.1&nbsp;</b>released
October 25th, 2017</p>

<p style="margin-left:32%;">Fixed bug with bad default
value of 1 instead of 32 for minseqlength when using the
makeudb_usearch command.</p>

<p style="margin-left:22%;"><b>v2.5.2&nbsp;</b>released
October 30th, 2017</p>

<p style="margin-left:32%;">Fixed bug with where
&rsquo;-&rsquo; as an argument to the fastq_eestats2 option
was treated literally instead of equivalent to stdin.</p>

<p style="margin-left:22%;"><b>v2.6.0&nbsp;</b>released
November 10th, 2017</p>

<p style="margin-left:32%;">Rewritten paired-end reads
merger with improved accuracy. Decreased default value for
fastq_minovlen option from 16 to 10. The default value for
the fastq_maxdiffs option is increased from 5 to 10. There
are now other more important restrictions that will avoid
merging reads that cannot be reliably aligned.</p>

<p style="margin-left:22%;"><b>v2.6.1&nbsp;</b>released
December 8th, 2017</p>

<p style="margin-left:32%;">Improved parallelisation of
paired end reads merging.</p>

<p style="margin-left:22%;"><b>v2.6.2&nbsp;</b>released
December 18th, 2017</p>

<p style="margin-left:32%;">Fixed option xsize that was
partially inactive for commands uchime_denovo, uchime_ref,
and fastx_filter.</p>

<p style="margin-left:22%;"><b>v2.7.0&nbsp;</b>released
February 13th, 2018</p>

<p style="margin-left:32%;">Added commands cluster_unoise,
uchime2_denovo and uchime3_denovo contributed by Davide
Albanese based on Robert Edgar&rsquo;s papers. Refactored
fasta and fastq print functions as well as code for
extraction of abundance and other attributes from the
headers.</p>

<p style="margin-left:22%;"><b>v2.7.1&nbsp;</b>released
February 16th, 2018</p>

<p style="margin-left:32%;">Fix several bugs on Windows
related to large files, use of &quot;-&quot; as a file name
to mean stdin or stdout, alignment errors, missed kmers and
corrupted UDB files. Added documentation of UDB-related
commands.</p>

<p style="margin-left:22%;"><b>v2.7.2&nbsp;</b>released
April 20th, 2018</p>

<p style="margin-left:32%;">Added the sintax command for
taxonomic classification. Fixed a bug with incorrect FASTA
headers of consensus sequences after clustering.</p>

<p style="margin-left:22%;"><b>v2.8.0&nbsp;</b>released
April 24th, 2018</p>

<p style="margin-left:32%;">Added the fastq_maxdiffpct
option to the fastq_mergepairs command.</p>

<p style="margin-left:22%;"><b>v2.8.1&nbsp;</b>released
June 22nd, 2018</p>

<p style="margin-left:32%;">Fixes for compilation warnings
with GCC 8.</p>

<p style="margin-left:22%;"><b>v2.8.2&nbsp;</b>released
August 21st, 2018</p>

<p style="margin-left:32%;">Fix for wrong placement of
semicolons in header lines in some cases when using the
sizeout or xsize options. Reduced memory requirements for
full-length dereplication in cases with many duplicate
sequences. Improved wording of fastq_mergepairs report.
Updated manual regarding use of sizein and sizeout with
dereplication. Changed a compiler option.</p>

<p style="margin-left:22%;"><b>v2.8.3&nbsp;</b>released
August 31st, 2018</p>

<p style="margin-left:32%;">Fix for segmentation fault for
--derep_fulllength with --uc.</p>

<p style="margin-left:22%;"><b>v2.8.4&nbsp;</b>released
September 3rd, 2018</p>

<p style="margin-left:32%;">Further reduce memory
requirements for dereplication when not using the uc option.
Fix output during subsampling when quiet or log options are
in effect.</p>

<p style="margin-left:22%;"><b>v2.8.5&nbsp;</b>released
September 26th, 2018</p>

<p style="margin-left:32%;">Fixed a bug in fastq_eestats2
that caused the values for large lengths to be much too high
when the input sequences had varying lengths.</p>

<p style="margin-left:22%;"><b>v2.8.6&nbsp;</b>released
October 9th, 2018</p>

<p style="margin-left:32%;">Fixed a bug introduced in
version 2.8.2 that caused derep_fulllength to include the
full FASTA header in its output instead of stopping at the
first space (unless the notrunclabels option is in
effect).</p>

<p style="margin-left:22%;"><b>v2.9.0&nbsp;</b>released
October 10th, 2018</p>

<p style="margin-left:32%;">Added the fastq_join
command.</p>

<p style="margin-left:22%;"><b>v2.9.1&nbsp;</b>released
October 29th, 2018</p>

<p style="margin-left:32%;">Changed compiler options that
select the target cpu and tuning to allow the software to
run on any 64-bit x86 system, while tuning for more modern
variants. Avoid illegal instruction error on some
architectures. Update documentation of rereplicate
command.</p>

<p style="margin-left:22%;"><b>v2.10.0&nbsp;</b>released
December 6th, 2018</p>

<p style="margin-left:32%;">Added the sff_convert commmand
to convert SFF files to FASTQ. Added some additional option
argument checks. Fixed segmentation fault bug after some
fatal errors when a log file was specified.</p>

<p style="margin-left:22%;"><b>v2.10.1&nbsp;</b>released
December 7th, 2018</p>

<p style="margin-left:32%;">Improved sff_convert command.
It will now read several variants of the SFF format. It is
also able to read from a pipe. Warnings are given if there
are minor problems. Errors messages have been improved.
Minor speed and memory usage improvements.</p>

<p style="margin-left:22%;"><b>v2.10.2&nbsp;</b>released
December 10th, 2018</p>

<p style="margin-left:32%;">Fixed bug in sintax with
reversed order of domain and kingdom.</p>

<p style="margin-left:22%;"><b>v2.10.3&nbsp;</b>released
December 19th, 2018</p>

<p style="margin-left:32%;">Ported to Linux on ARMv8
(aarch64). Fixed compilation warning with gcc version 8.1.0
and 8.2.0.</p>

<p style="margin-left:22%;"><b>v2.10.4&nbsp;</b>released
January 4th, 2019</p>

<p style="margin-left:32%;">Fixed serious bug in x86_64
SIMD alignment code introduced in version 2.10.3. Added link
to BioConda in README. Fixed bug in fastq_stats with
sequence length 1. Fixed use of equals symbol in UC files
for identical sequences with cluster_fast.</p>

<p style="margin-left:22%;"><b>v2.11.0&nbsp;</b>released
February 13th, 2019</p>

<p style="margin-left:32%;">Added ability to trim and
filter paired-end reads using the reverse option with the
fastx_filter and fastq_filter commands. Added --xee option
to remove ee attributes from FASTA headers. Minor invisible
improvement to the progress indicator.</p>

<p style="margin-left:22%;"><b>v2.11.1&nbsp;</b>released
February 28th, 2019</p>

<p style="margin-left:32%;">Minor change to the handling of
the weak_id and id options when using cluster_unoise.</p>

<p style="margin-left:22%;"><b>v2.12.0&nbsp;</b>released
March 19th, 2019</p>

<p style="margin-left:32%;">Take sequence abundance into
account when computing consensus sequences or profiles after
clustering. Warn when rereplicating sequences without
abundance info. Guess offset 33 in more cases with
fastq_chars. Stricter checking of option arguments and
option combinations.</p>

<p style="margin-left:22%;"><b>v2.13.0&nbsp;</b>released
April 11th, 2019</p>

<p style="margin-left:32%;">Added the --fastx_getseq,
--fastx_getseqs and --fastx_getsubseq commands to extract
sequences from a FASTA or FASTQ file based on their labels.
Improved handling of ambiguous nucleotide symbols. Corrected
behaviour of --uchime_ref command with and options --self
and --selfid. Strict detection of illegal options for each
command.</p>

<p style="margin-left:22%;"><b>v2.13.1&nbsp;</b>released
April 26th, 2019</p>

<p style="margin-left:32%;">Minor changes to the allowed
options for each command. All commands now allow the log,
quiet and threads options. If more than 1 thread is
specified for commands that are not multi-threaded, a
warning will be issued. Minor changes to the manual.</p>

<p style="margin-left:22%;"><b>v2.13.2&nbsp;</b>released
April 30th, 2019</p>

<p style="margin-left:32%;">Fixed bug related to improper
handling of newlines on Windows. Allowed option strand plus
to uchime_ref for compatibility.</p>

<p style="margin-left:22%;"><b>v2.13.3&nbsp;</b>released
April 30th, 2019</p>

<p style="margin-left:32%;">Fixed bug in FASTQ parsing
introduced in version 2.13.2.</p>

<p style="margin-left:22%;"><b>v2.13.4&nbsp;</b>released
May 10th, 2019</p>

<p style="margin-left:32%;">Added information about support
for gzip- and bzip2-compressed input files to the output of
the version command. Adapted source code for compilation on
FreeBSD and NetBSD systems.</p>

<p style="margin-left:22%;"><b>v2.13.5&nbsp;</b>released
July 2nd, 2019</p>

<p style="margin-left:32%;">Added cut command to fragment
sequences at restriction sites. Silenced output from the
fastq_stats command if quiet option was given. Updated
manual.</p>

<p style="margin-left:22%;"><b>v2.13.6&nbsp;</b>released
July 2nd, 2019</p>

<p style="margin-left:32%;">Added info about cut command to
output of help command.</p>
<hr>
</body>
</html>
